diff -ruN libmikmod-3.2.0/drivers/drv_raw.c libmikmod-3.2.0.aros/drivers/drv_raw.c
--- libmikmod-3.2.0/drivers/drv_raw.c	2012-03-14 17:02:59.000000000 +0000
+++ libmikmod-3.2.0.aros/drivers/drv_raw.c	2012-06-10 19:11:45.266629400 +0100
@@ -51,6 +51,11 @@
 #define O_BINARY 0
 #endif
 
+#ifndef S_IREAD
+#	define S_IREAD 0
+#	define S_IWRITE 1
+#endif
+
 #if defined(WIN32) && !defined(__MWERKS__)
 #define open _open
 #endif
diff -ruN libmikmod-3.2.0/include/mikmod.h libmikmod-3.2.0.aros/include/mikmod.h
--- libmikmod-3.2.0/include/mikmod.h	2012-06-01 14:15:42.000000000 +0100
+++ libmikmod-3.2.0.aros/include/mikmod.h	2012-06-10 19:11:45.521644000 +0100
@@ -104,7 +104,7 @@
 typedef signed short    SWORD;      /* 2 bytes, signed */
 typedef unsigned short  UWORD;      /* 2 bytes, unsigned */
 typedef signed long     SLONG;      /* 4 bytes, signed */
-#if !defined(__OS2__)&&!defined(__EMX__)&&!defined(WIN32)
+#if (defined(__AROS__)&&!defined(EXEC_TYPES_H))&&!defined(__OS2__)&&!defined(__EMX__)&&!defined(WIN32)
 typedef unsigned long   ULONG;      /* 4 bytes, unsigned */
 typedef int             BOOL;       /* 0=false, <>0 true */
 #endif
diff -ruN libmikmod-3.2.0/include/mikmod.h.in libmikmod-3.2.0.aros/include/mikmod.h.in
--- libmikmod-3.2.0/include/mikmod.h.in	2012-05-12 10:52:13.000000000 +0100
+++ libmikmod-3.2.0.aros/include/mikmod.h.in	2012-06-10 19:11:45.518643800 +0100
@@ -104,7 +104,7 @@
 typedef signed short    SWORD;      /* 2 bytes, signed */
 typedef unsigned short  UWORD;      /* 2 bytes, unsigned */
 typedef signed long     SLONG;      /* 4 bytes, signed */
-#if !defined(__OS2__)&&!defined(__EMX__)&&!defined(WIN32)
+#if (defined(__AROS__)&&!defined(EXEC_TYPES_H))&&!defined(__OS2__)&&!defined(__EMX__)&&!defined(WIN32)
 typedef unsigned long   ULONG;      /* 4 bytes, unsigned */
 typedef int             BOOL;       /* 0=false, <>0 true */
 #endif
