diff -ruN blobwars-1.19/src/CReplayData.cpp blobwars-1.19.aros/src/CReplayData.cpp
--- blobwars-1.19/src/CReplayData.cpp	2011-04-17 15:56:56.000000000 +0100
+++ blobwars-1.19.aros/src/CReplayData.cpp	2014-08-01 12:40:12.222154071 +0100
@@ -38,7 +38,11 @@
 
 void ReplayData::printReplayInformation()
 {
+#ifdef __AROS__
+	tm *timeinfo = localtime((const time_t*)&header.randomSeed);
+#else
 	tm *timeinfo = localtime(&header.randomSeed);
+#endif
 	printf("Recorded on : %s", asctime(timeinfo));
 	printf("Map         : %s\n", header.map);
 	printf("Score       : %d\n", header.score);
diff -ruN blobwars-1.19/src/headers.h blobwars-1.19.aros/src/headers.h
--- blobwars-1.19/src/headers.h	2011-04-17 15:56:56.000000000 +0100
+++ blobwars-1.19.aros/src/headers.h	2014-08-01 12:40:12.222154071 +0100
@@ -35,6 +35,12 @@
 #ifndef SDL_GetGamma
 extern DECLSPEC int SDLCALL SDL_GetGamma(float *red, float *green, float *blue);
 #endif
+#elif __AROS__
+#include <SDL/SDL.h>
+#include <SDL/SDL_image.h>
+#include <SDL/SDL_mixer.h>
+#include <SDL/SDL_ttf.h>
+#include <SDL/SDL_net.h>
 #else
 #include "SDL/SDL.h"
 #include "SDL/SDL_image.h"
@@ -43,7 +49,7 @@
 #include "SDL/SDL_net.h"
 #endif
 
-#ifndef WIN32
+#if !defined(WIN32) && !defined(__AROS__)
 #include <libintl.h>
 #define _(string) gettext(string)
 #else
diff -ruN blobwars-1.19/src/init.cpp blobwars-1.19.aros/src/init.cpp
--- blobwars-1.19/src/init.cpp	2011-04-17 15:56:56.000000000 +0100
+++ blobwars-1.19.aros/src/init.cpp	2014-08-01 12:40:12.222154071 +0100
@@ -92,6 +92,12 @@
 #if UNIX
 void setupUserHomeDirectory()
 {
+	char dir[PATH_MAX];
+	dir[0] = 0;
+
+#ifdef __AROS__
+	char userHome[] = "/PROGDIR";
+#else
 	char *userHome = getenv("HOME");
 	
 	if ((!userHome) || (userHome == NULL))
@@ -99,11 +105,9 @@
 		printf("Couldn't determine user home directory! Exitting.\n");
 		exit(1);
 	}
+#endif
 
 	debug(("User Home = %s\n", userHome));
-	
-	char dir[PATH_MAX];
-	dir[0] = 0;
 
 	snprintf(dir, sizeof dir, "%s/.parallelrealities", userHome);
 	if ((mkdir(dir, S_IRWXU|S_IRWXG|S_IROTH|S_IXOTH) != 0) && (errno != EEXIST))
diff -ruN blobwars-1.19/src/mapEditor.cpp blobwars-1.19.aros/src/mapEditor.cpp
--- blobwars-1.19/src/mapEditor.cpp	2011-04-17 15:56:56.000000000 +0100
+++ blobwars-1.19.aros/src/mapEditor.cpp	2014-08-01 12:42:53.566867275 +0100
@@ -242,7 +242,7 @@
 		fclose(fp);
 	}
 	
-	printf("Saved %s (%d)\n", name, SDL_GetTicks());
+	printf("Saved %s (%d)\n", name, (int)SDL_GetTicks());
 }
 
 void collectMapData()
