diff -Naur cflow-1.5-orig/src/rc.c cflow-1.5/src/rc.c
--- cflow-1.5-orig/src/rc.c	2017-08-01 00:22:09.440756945 +0200
+++ cflow-1.5/src/rc.c	2017-08-01 00:26:02.641823593 +0200
@@ -115,7 +115,11 @@
      if (env) 
 	  parse_rc(&xargc, &xargv, env);
      else {
+#if defined(__AROS__)
+	  char *home = "PROGDIR:";
+#else
 	  char *home = getenv("HOME");
+#endif
 	  if (home) {
 	       int len = strlen(home);
 	       char *buf = malloc(len + sizeof(LOCAL_RC)
@@ -123,8 +127,10 @@
 	       if (!buf)
 		    return;
 	       strcpy(buf, home);
+#if !defined(__AROS__)
 	       if (home[len-1] != '/')
 		    buf[len++] = '/';
+#endif
 	       strcpy(buf+len, LOCAL_RC);
 	       parse_rc(&xargc, &xargv, buf);
 	       free(buf);
