diff -Naur cflow-1.4-orig/src/rc.c cflow-1.4/src/rc.c
--- cflow-1.4-orig/src/rc.c	2011-12-17 14:39:31.769772890 +0100
+++ cflow-1.4/src/rc.c	2011-12-17 14:42:33.179781604 +0100
@@ -108,7 +108,11 @@
      if (env) 
 	  parse_rc(&xargc, &xargv, env);
      else {
+#if defined(__AROS__)
+	  char *home = "PROGDIR:";
+#else
 	  char *home = getenv("HOME");
+#endif
 	  if (home) {
 	       int len = strlen(home);
 	       char *buf = malloc(len + sizeof(LOCAL_RC)
@@ -116,8 +120,10 @@
 	       if (!buf)
 		    return;
 	       strcpy(buf, home);
+#if !defined(__AROS__)
 	       if (home[len-1] != '/')
 		    buf[len++] = '/';
+#endif
 	       strcpy(buf+len, LOCAL_RC);
 	       parse_rc(&xargc, &xargv, buf);
 	       free(buf);
