diff -ruN naev-0.7.0/lib/lua/luaconf.h naev-0.7.0.aros/lib/lua/luaconf.h
--- naev-0.7.0/lib/lua/luaconf.h	2017-07-16 00:51:22.000000000 +0200
+++ naev-0.7.0.aros/lib/lua/luaconf.h	2018-07-29 02:46:05.377669697 +0200
@@ -55,7 +55,7 @@
 #define LUA_USE_MKSTEMP
 #define LUA_USE_ISATTY
 #define LUA_USE_POPEN
-#define LUA_USE_ULONGJMP
+//#define LUA_USE_ULONGJMP
 #endif
 
 
diff -ruN naev-0.7.0/src/naev.c naev-0.7.0.aros/src/naev.c
--- naev-0.7.0/src/naev.c	2017-07-16 00:51:22.000000000 +0200
+++ naev-0.7.0.aros/src/naev.c	2018-07-28 11:55:34.255841964 +0200
@@ -884,7 +884,7 @@
  */
 static void fps_init (void)
 {
-#if HAS_POSIX && defined(CLOCK_MONOTONIC)
+#if HAS_POSIX && defined(CLOCK_MONOTONIC) && !defined(__AROS__)
    use_posix_time = 1;
    /* We must use clock_gettime here instead of gettimeofday mainly because this
     * way we are not influenced by changes to the time source like say ntp which
@@ -906,7 +906,7 @@
    double dt;
    unsigned int t;
 
-#if HAS_POSIX && defined(CLOCK_MONOTONIC)
+#if HAS_POSIX && defined(CLOCK_MONOTONIC) && !defined(__AROS__)
    struct timespec ts;
 
    if (use_posix_time) {
diff -ruN naev-0.7.0/src/ndata.c naev-0.7.0.aros/src/ndata.c
--- naev-0.7.0/src/ndata.c	2017-07-16 00:51:22.000000000 +0200
+++ naev-0.7.0.aros/src/ndata.c	2018-07-28 11:55:34.255841964 +0200
@@ -209,7 +209,11 @@
 #endif /* SDL_VERSION_ATLEAST(2,0,0) */
 
    /* Create the surface. */
+#if SDL_VERSION_ATLEAST(2,0,0)
    rw    = SDL_RWFromConstMem( nondata_png, sizeof(nondata_png) );
+#else
+   rw    = SDL_RWFromMem( (void *)nondata_png, sizeof(nondata_png) );
+#endif /* SDL_VERSION_ATLEAST(2,0,0) */
    npng  = npng_open( rw );
    sur   = npng_readSurface( npng, 0, 0 );
    npng_close( npng );
