diff -ruN lbreakout2-2.6.5/Makefile.am lbreakout2-2.6.5.aros/Makefile.am
--- lbreakout2-2.6.5/Makefile.am	2013-05-03 18:06:20.000000000 +0100
+++ lbreakout2-2.6.5.aros/Makefile.am	2015-10-17 17:10:12.000000000 +0100
@@ -21,7 +21,7 @@
 # does not contain plus signs. :-)
 lbreakout2.desktop:
 	@sed -e "s+Icon=+Icon=$(datadir)/icons/lbreakout48.gif+" \
-	< lbreakout2.desktop.in > lbreakout2.desktop
+	< $(srcdir)/lbreakout2.desktop.in > lbreakout2.desktop
 
 win32text: $(WIN32_TEXT)
 	@echo Converting text files for win32 installer
diff -ruN lbreakout2-2.6.5/Makefile.in lbreakout2-2.6.5.aros/Makefile.in
--- lbreakout2-2.6.5/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/Makefile.in	2015-10-17 17:10:12.000000000 +0100
@@ -727,7 +727,7 @@
 # does not contain plus signs. :-)
 lbreakout2.desktop:
 	@sed -e "s+Icon=+Icon=$(datadir)/icons/lbreakout48.gif+" \
-	< lbreakout2.desktop.in > lbreakout2.desktop
+	< $(srcdir)/lbreakout2.desktop.in > lbreakout2.desktop
 
 win32text: $(WIN32_TEXT)
 	@echo Converting text files for win32 installer
diff -ruN lbreakout2-2.6.5/client/Makefile.am lbreakout2-2.6.5.aros/client/Makefile.am
--- lbreakout2-2.6.5/client/Makefile.am	2013-05-03 18:06:20.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/Makefile.am	2015-10-17 17:10:12.000000000 +0100
@@ -51,7 +51,7 @@
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)
 	$(mkinstalldirs) $(DESTDIR)$(hi_dir)
 	if ! test -f $(DESTDIR)$(hi_dir)/lbreakout2.hscr; then \
-	  $(INSTALL_DATA) -m 666 empty.hscr $(DESTDIR)$(hi_dir)/lbreakout2.hscr; \
+	  $(INSTALL_DATA) -m 666 $(srcdir)/empty.hscr $(DESTDIR)$(hi_dir)/lbreakout2.hscr; \
 	fi;
 
 uninstall-local:
diff -ruN lbreakout2-2.6.5/client/Makefile.in lbreakout2-2.6.5.aros/client/Makefile.in
--- lbreakout2-2.6.5/client/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/Makefile.in	2015-10-17 17:10:12.000000000 +0100
@@ -670,7 +670,7 @@
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)
 	$(mkinstalldirs) $(DESTDIR)$(hi_dir)
 	if ! test -f $(DESTDIR)$(hi_dir)/lbreakout2.hscr; then \
-	  $(INSTALL_DATA) -m 666 empty.hscr $(DESTDIR)$(hi_dir)/lbreakout2.hscr; \
+	  $(INSTALL_DATA) -m 666 $(srcdir)/empty.hscr $(DESTDIR)$(hi_dir)/lbreakout2.hscr; \
 	fi;
 
 uninstall-local:
diff -ruN lbreakout2-2.6.5/client/gfx/AbsoluteB/Makefile.am lbreakout2-2.6.5.aros/client/gfx/AbsoluteB/Makefile.am
--- lbreakout2-2.6.5/client/gfx/AbsoluteB/Makefile.am	2013-05-03 18:06:14.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gfx/AbsoluteB/Makefile.am	2015-10-17 17:10:12.000000000 +0100
@@ -6,7 +6,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gfx/AbsoluteB
 	@for file in $(EXTRA_DIST); do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gfx/AbsoluteB/$$file;\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gfx/AbsoluteB/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/gfx/AbsoluteB/Makefile.in lbreakout2-2.6.5.aros/client/gfx/AbsoluteB/Makefile.in
--- lbreakout2-2.6.5/client/gfx/AbsoluteB/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gfx/AbsoluteB/Makefile.in	2015-10-17 17:10:12.000000000 +0100
@@ -360,7 +360,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gfx/AbsoluteB
 	@for file in $(EXTRA_DIST); do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gfx/AbsoluteB/$$file;\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gfx/AbsoluteB/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/gfx/Classic/Makefile.am lbreakout2-2.6.5.aros/client/gfx/Classic/Makefile.am
--- lbreakout2-2.6.5/client/gfx/Classic/Makefile.am	2013-05-03 18:06:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gfx/Classic/Makefile.am	2015-10-17 17:10:12.000000000 +0100
@@ -3,7 +3,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gfx/Classic
 	@for file in $(EXTRA_DIST); do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gfx/Classic/$$file;\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gfx/Classic/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/gfx/Classic/Makefile.in lbreakout2-2.6.5.aros/client/gfx/Classic/Makefile.in
--- lbreakout2-2.6.5/client/gfx/Classic/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gfx/Classic/Makefile.in	2015-10-17 17:10:12.000000000 +0100
@@ -356,7 +356,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gfx/Classic
 	@for file in $(EXTRA_DIST); do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gfx/Classic/$$file;\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gfx/Classic/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/gfx/Makefile.am lbreakout2-2.6.5.aros/client/gfx/Makefile.am
--- lbreakout2-2.6.5/client/gfx/Makefile.am	2013-05-03 18:06:19.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gfx/Makefile.am	2015-10-17 17:10:12.000000000 +0100
@@ -19,7 +19,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gfx
 	@for file in $(EXTRA_DIST); do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gfx/$$file;\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gfx/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/gfx/Makefile.in lbreakout2-2.6.5.aros/client/gfx/Makefile.in
--- lbreakout2-2.6.5/client/gfx/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gfx/Makefile.in	2015-10-17 17:10:12.000000000 +0100
@@ -529,7 +529,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gfx
 	@for file in $(EXTRA_DIST); do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gfx/$$file;\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gfx/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/gfx/Moiree/Makefile.am lbreakout2-2.6.5.aros/client/gfx/Moiree/Makefile.am
--- lbreakout2-2.6.5/client/gfx/Moiree/Makefile.am	2013-05-03 18:06:10.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gfx/Moiree/Makefile.am	2015-10-17 17:10:12.000000000 +0100
@@ -7,7 +7,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gfx/Moiree
 	@for file in $(EXTRA_DIST); do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gfx/Moiree/$$file;\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gfx/Moiree/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/gfx/Moiree/Makefile.in lbreakout2-2.6.5.aros/client/gfx/Moiree/Makefile.in
--- lbreakout2-2.6.5/client/gfx/Moiree/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gfx/Moiree/Makefile.in	2015-10-17 17:10:12.000000000 +0100
@@ -361,7 +361,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gfx/Moiree
 	@for file in $(EXTRA_DIST); do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gfx/Moiree/$$file;\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gfx/Moiree/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/gfx/Oz/Makefile.am lbreakout2-2.6.5.aros/client/gfx/Oz/Makefile.am
--- lbreakout2-2.6.5/client/gfx/Oz/Makefile.am	2013-05-03 18:06:16.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gfx/Oz/Makefile.am	2015-10-17 17:10:12.000000000 +0100
@@ -6,7 +6,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gfx/Oz
 	@for file in $(EXTRA_DIST); do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gfx/Oz/$$file;\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gfx/Oz/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/gfx/Oz/Makefile.in lbreakout2-2.6.5.aros/client/gfx/Oz/Makefile.in
--- lbreakout2-2.6.5/client/gfx/Oz/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gfx/Oz/Makefile.in	2015-10-17 17:10:12.000000000 +0100
@@ -360,7 +360,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gfx/Oz
 	@for file in $(EXTRA_DIST); do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gfx/Oz/$$file;\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gfx/Oz/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/gui_theme/Makefile.am lbreakout2-2.6.5.aros/client/gui_theme/Makefile.am
--- lbreakout2-2.6.5/client/gui_theme/Makefile.am	2013-05-03 18:06:10.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gui_theme/Makefile.am	2015-10-17 17:10:12.000000000 +0100
@@ -9,7 +9,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gui_theme
 	@for file in $(EXTRA_DIST); do \
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gui_theme/$$file; \
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gui_theme/$$file; \
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/gui_theme/Makefile.in lbreakout2-2.6.5.aros/client/gui_theme/Makefile.in
--- lbreakout2-2.6.5/client/gui_theme/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/gui_theme/Makefile.in	2015-10-17 17:10:12.000000000 +0100
@@ -363,7 +363,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/gui_theme
 	@for file in $(EXTRA_DIST); do \
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/gui_theme/$$file; \
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/gui_theme/$$file; \
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/levels/Makefile.am lbreakout2-2.6.5.aros/client/levels/Makefile.am
--- lbreakout2-2.6.5/client/levels/Makefile.am	2014-12-20 13:48:34.000000000 +0000
+++ lbreakout2-2.6.5.aros/client/levels/Makefile.am	2015-10-17 17:10:12.000000000 +0100
@@ -138,7 +138,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/levels
 	@for file in $(EXTRA_DIST); do \
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/levels/$$file; \
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/levels/$$file; \
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/levels/Makefile.in lbreakout2-2.6.5.aros/client/levels/Makefile.in
--- lbreakout2-2.6.5/client/levels/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/levels/Makefile.in	2015-10-17 17:10:12.000000000 +0100
@@ -471,7 +471,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/levels
 	@for file in $(EXTRA_DIST); do \
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/levels/$$file; \
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/levels/$$file; \
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/sounds/Makefile.am lbreakout2-2.6.5.aros/client/sounds/Makefile.am
--- lbreakout2-2.6.5/client/sounds/Makefile.am	2013-05-03 18:06:20.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/sounds/Makefile.am	2019-04-07 01:28:20.611666800 +0100
@@ -8,8 +8,8 @@
 
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/sounds
-	@for file in *.wav; do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/sounds/$$file;\
+	@for file in $(EXTRA_DIST); do\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/sounds/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/client/sounds/Makefile.in lbreakout2-2.6.5.aros/client/sounds/Makefile.in
--- lbreakout2-2.6.5/client/sounds/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/client/sounds/Makefile.in	2019-04-07 01:28:02.037655800 +0100
@@ -362,8 +362,8 @@
 
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/sounds
-	@for file in *.wav; do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/sounds/$$file;\
+	@for file in $(EXTRA_DIST); do\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/sounds/$$file;\
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/docs/Makefile.am lbreakout2-2.6.5.aros/docs/Makefile.am
--- lbreakout2-2.6.5/docs/Makefile.am	2013-05-03 18:06:05.000000000 +0100
+++ lbreakout2-2.6.5.aros/docs/Makefile.am	2019-04-07 01:29:04.829254900 +0100
@@ -8,10 +8,9 @@
 
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(doc_dir)/lbreakout2
-	@for file in *.jpg; do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(doc_dir)/lbreakout2/$$file;\
+	@for file in $(EXTRA_DIST); do\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(doc_dir)/lbreakout2/$$file;\
 	done
-	$(INSTALL_DATA) index.html $(DESTDIR)$(doc_dir)/lbreakout2/index.html
 
 win32-install-script:
 	@for file in $(EXTRA_DIST); do \
diff -ruN lbreakout2-2.6.5/docs/Makefile.in lbreakout2-2.6.5.aros/docs/Makefile.in
--- lbreakout2-2.6.5/docs/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/docs/Makefile.in	2019-04-07 01:29:30.426115100 +0100
@@ -362,10 +362,9 @@
 
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(doc_dir)/lbreakout2
-	@for file in *.jpg; do\
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(doc_dir)/lbreakout2/$$file;\
+	@for file in $(EXTRA_DIST); do\
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(doc_dir)/lbreakout2/$$file;\
 	done
-	$(INSTALL_DATA) index.html $(DESTDIR)$(doc_dir)/lbreakout2/index.html
 
 win32-install-script:
 	@for file in $(EXTRA_DIST); do \
diff -ruN lbreakout2-2.6.5/server/levels/Makefile.am lbreakout2-2.6.5.aros/server/levels/Makefile.am
--- lbreakout2-2.6.5/server/levels/Makefile.am	2013-05-03 18:06:06.000000000 +0100
+++ lbreakout2-2.6.5.aros/server/levels/Makefile.am	2015-10-17 17:10:12.000000000 +0100
@@ -3,7 +3,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/levels
 	@for file in $(EXTRA_DIST); do \
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/levels/$$file; \
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/levels/$$file; \
 	done
 
 win32-install-script:
diff -ruN lbreakout2-2.6.5/server/levels/Makefile.in lbreakout2-2.6.5.aros/server/levels/Makefile.in
--- lbreakout2-2.6.5/server/levels/Makefile.in	2015-10-17 17:10:11.000000000 +0100
+++ lbreakout2-2.6.5.aros/server/levels/Makefile.in	2015-10-17 17:10:12.000000000 +0100
@@ -356,7 +356,7 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(inst_dir)/levels
 	@for file in $(EXTRA_DIST); do \
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(inst_dir)/levels/$$file; \
+	  $(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(inst_dir)/levels/$$file; \
 	done
 
 win32-install-script:
