diff -urN cairo-1.10.2//configure cairo-1.10.2-aros//configure
--- cairo-1.10.2//configure	2010-12-25 15:22:57.000000000 +0100
+++ cairo-1.10.2-aros//configure	2012-03-31 14:34:52.000000000 +0200
@@ -26453,7 +26453,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_png_CFLAGS=`$PKG_CONFIG --cflags "$png_REQUIRES" 2>/dev/null`
+  pkg_cv_png_CFLAGS=""
 else
   pkg_failed=yes
 fi
@@ -26469,7 +26469,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_png_LIBS=`$PKG_CONFIG --libs "$png_REQUIRES" 2>/dev/null`
+  pkg_cv_png_LIBS=""
 else
   pkg_failed=yes
 fi
@@ -29329,7 +29329,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_FREETYPE_CFLAGS=`$PKG_CONFIG --cflags "freetype2 >= $FREETYPE_MIN_VERSION" 2>/dev/null`
+  pkg_cv_FREETYPE_CFLAGS=""
 else
   pkg_failed=yes
 fi
@@ -29345,7 +29345,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_FREETYPE_LIBS=`$PKG_CONFIG --libs "freetype2 >= $FREETYPE_MIN_VERSION" 2>/dev/null`
+  pkg_cv_FREETYPE_LIBS=""
 else
   pkg_failed=yes
 fi
@@ -29777,7 +29777,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_FONTCONFIG_CFLAGS=`$PKG_CONFIG --cflags "$fc_REQUIRES" 2>/dev/null`
+  pkg_cv_FONTCONFIG_CFLAGS=""
 else
   pkg_failed=yes
 fi
@@ -29793,7 +29793,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_FONTCONFIG_LIBS=`$PKG_CONFIG --libs "$fc_REQUIRES" 2>/dev/null`
+  pkg_cv_FONTCONFIG_LIBS=""
 else
   pkg_failed=yes
 fi
@@ -31616,7 +31616,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_pixman_CFLAGS=`$PKG_CONFIG --cflags "$pixman_REQUIRES" 2>/dev/null`
+  pkg_cv_pixman_CFLAGS=""
 else
   pkg_failed=yes
 fi
@@ -31632,7 +31632,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_pixman_LIBS=`$PKG_CONFIG --libs "$pixman_REQUIRES" 2>/dev/null`
+  pkg_cv_pixman_LIBS=""
 else
   pkg_failed=yes
 fi
diff -urN cairo-1.10.2//src/cairo-compiler-private.h cairo-1.10.2-aros//src/cairo-compiler-private.h
--- cairo-1.10.2//src/cairo-compiler-private.h	2010-06-18 13:47:12.000000000 +0200
+++ cairo-1.10.2-aros//src/cairo-compiler-private.h	2012-03-31 14:57:11.000000000 +0200
@@ -229,6 +229,10 @@
 
 #endif
 
+#if defined(__AROS__)
+#define ffs __builtin_ffs
+#endif
+
 #if defined(_MSC_VER) && defined(_M_IX86)
 /* When compiling with /Gy and /OPT:ICF identical functions will be folded in together.
    The CAIRO_ENSURE_UNIQUE macro ensures that a function is always unique and
diff -urN cairo-1.10.2//src/cairo-mutex-impl-private.h cairo-1.10.2-aros//src/cairo-mutex-impl-private.h
--- cairo-1.10.2//src/cairo-mutex-impl-private.h	2010-06-18 13:47:12.000000000 +0200
+++ cairo-1.10.2-aros//src/cairo-mutex-impl-private.h	2012-03-31 14:52:44.000000000 +0200
@@ -156,6 +156,10 @@
  *   poke around cairo-mutex-private.h for possible solutions.
  */
 
+#if defined(__AROS__)
+#define CAIRO_NO_MUTEX 1
+#endif
+
 #if CAIRO_NO_MUTEX
 
 /* No mutexes */
diff -urN cairo-1.10.2//src/pixman.h cairo-1.10.2-aros//src/pixman.h
--- cairo-1.10.2//src/pixman.h	1970-01-01 01:00:00.000000000 +0100
+++ cairo-1.10.2-aros//src/pixman.h	2012-03-31 15:49:52.000000000 +0200
@@ -0,0 +1,2 @@
+#include <pixman-1/pixman.h>
+
diff -urN cairo-1.10.2//src/pixman-version.h cairo-1.10.2-aros//src/pixman-version.h
--- cairo-1.10.2//src/pixman-version.h	1970-01-01 01:00:00.000000000 +0100
+++ cairo-1.10.2-aros//src/pixman-version.h	2012-03-31 15:50:15.000000000 +0200
@@ -0,0 +1,2 @@
+#include <pixman-1/pixman-version.h>
+
