diff -u -r fileutils-4.1/config.sub fileutils-4.1-aros/config.sub
--- fileutils-4.1/config.sub	Sat Apr 21 14:27:49 2001
+++ fileutils-4.1-aros/config.sub	Tue Apr 27 19:24:25 2004
@@ -326,7 +326,11 @@
 		basic_machine=m68k-unknown
 		os=-sysv4
 		;;
-	apollo68)
+	aros)
+		basic_machine=i386-unknown
+		os=-aros
+		;;
+		apollo68)
 		basic_machine=m68k-apollo
 		os=-sysv
 		;;
@@ -1009,7 +1013,7 @@
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -solaris* | -sym* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* \
+	      | -aos* | -aros* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -netbsd* | -openbsd* | -freebsd* | -riscix* \
diff -u -r fileutils-4.1/configure fileutils-4.1-aros/configure
--- fileutils-4.1/configure	Sun Apr 29 04:26:09 2001
+++ fileutils-4.1-aros/configure	Tue Apr 27 19:24:25 2004
@@ -17797,7 +17797,7 @@
 if test "${ac_cv_func_setvbuf_reversed+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = foo; then
   { { echo "$as_me:17801: error: cannot run test program while cross compiling" >&5
 echo "$as_me: error: cannot run test program while cross compiling" >&2;}
    { (exit 1); exit 1; }; }
diff -u -r fileutils-4.1/lib/getloadavg.c fileutils-4.1-aros/lib/getloadavg.c
--- fileutils-4.1/lib/getloadavg.c	Mon Jun 19 03:22:00 2000
+++ fileutils-4.1-aros/lib/getloadavg.c	Tue Apr 27 19:24:25 2004
@@ -70,7 +70,7 @@
 
    We also #define LDAV_PRIVILEGED if a program will require
    special installation to be able to call getloadavg.  */
-
+#ifndef __AROS__
 /* This should always be first.  */
 #ifdef HAVE_CONFIG_H
 # include <config.h>
@@ -1084,3 +1084,4 @@
   exit (0);
 }
 #endif /* TEST */
+#endif /* __AROS__ */
diff -u -r fileutils-4.1/src/Makefile.in fileutils-4.1-aros/src/Makefile.in
--- fileutils-4.1/src/Makefile.in	Sun Apr 29 07:40:37 2001
+++ fileutils-4.1-aros/src/Makefile.in	Tue Apr 27 19:24:25 2004
@@ -122,9 +122,16 @@
 EXTRA_PROGRAMS = df
 
 bin_PROGRAMS = chgrp chown chmod cp dd dircolors du \
-ginstall ln dir vdir ls mkdir \
-mkfifo mknod mv rm rmdir shred sync touch @DF_PROG@
+dir vdir ls mkdir \
+mkfifo mknod mv rm rmdir touch @DF_PROG@
 
+# AROS - I pulled the following programs from the above list:
+#  ginstall - exec/fork stuff is needed.
+#  shred - Not sure what this is, not a high priority, needs to be 
+#	   looked at
+#  sync - Not looked at all. Not critical atm.
+#  ln - Useless without addressing link() and symlink() maybe fake the 
+#	functions by copying files?
 
 datadir = $(prefix)/@DATADIRNAME@
 localedir = $(datadir)/locale
@@ -174,10 +181,16 @@
 CONFIG_CLEAN_FILES =
 EXTRA_PROGRAMS = df$(EXEEXT)
 bin_PROGRAMS = chgrp$(EXEEXT) chown$(EXEEXT) chmod$(EXEEXT) cp$(EXEEXT) \
-dd$(EXEEXT) dircolors$(EXEEXT) du$(EXEEXT) ginstall$(EXEEXT) \
-ln$(EXEEXT) dir$(EXEEXT) vdir$(EXEEXT) ls$(EXEEXT) mkdir$(EXEEXT) \
+dd$(EXEEXT) dircolors$(EXEEXT) du$(EXEEXT)  \
+dir$(EXEEXT) vdir$(EXEEXT) ls$(EXEEXT) mkdir$(EXEEXT) \
 mkfifo$(EXEEXT) mknod$(EXEEXT) mv$(EXEEXT) rm$(EXEEXT) rmdir$(EXEEXT) \
-shred$(EXEEXT) sync$(EXEEXT) touch$(EXEEXT) @DF_PROG@
+touch$(EXEEXT) @DF_PROG@
+
+# AROS - Removed from above list:
+# ginstall$(EXEEXT)
+# ln$(EXEEXT)
+# shred$(EXEEXT) sync$(EXEEXT)  
+
 PROGRAMS = $(bin_PROGRAMS)
 
 CPPFLAGS = @CPPFLAGS@
diff -u -r fileutils-4.1/src/copy.c fileutils-4.1-aros/src/copy.c
--- fileutils-4.1/src/copy.c	Sat Jan 13 10:49:43 2001
+++ fileutils-4.1-aros/src/copy.c	Tue Apr 27 19:24:25 2004
@@ -1217,3 +1217,14 @@
   return copy_internal (src_path, dst_path, nonexistent_dst, 0, NULL,
 			options, move_mode, copy_into_self, rename_succeeded);
 }
+/* AROS */
+
+/* This function is not in AROS libc. Probably the best thing outside of 
+implementing an actual function here would be just to copy the files, 
+but for now I'll return Operation not supported by file system, which is 
+truthful. */
+
+int link(const char *name1, const char *name2) {
+return EOPNOTSUPP;
+}
+
diff -u -r fileutils-4.1/src/cp.c fileutils-4.1-aros/src/cp.c
--- fileutils-4.1/src/cp.c	Sat Feb  3 11:48:34 2001
+++ fileutils-4.1-aros/src/cp.c	Tue Apr 27 19:24:25 2004
@@ -909,3 +909,20 @@
 
   exit (exit_status);
 }
+/* AROS */
+
+/* This is not in the aros libc, as aros has no special device files 
+atm. For the purposes of this program, we don't need it and won't use it 
+or care about the results (well hosted ppl might), therefore, return an 
+error. */
+
+int mknod(const char *path, mode_t mode, dev_t dev) {
+return -1;
+}
+
+/* This is not in AROS libc. It could copy the files, but I want to 
+test, so I'll just return not supported or an error. */
+int symlink(const char *name1, const char *name2) {
+return -1;
+}
+
diff -u -r fileutils-4.1/src/dd.c fileutils-4.1-aros/src/dd.c
--- fileutils-4.1/src/dd.c	Sun Apr 22 09:21:58 2001
+++ fileutils-4.1-aros/src/dd.c	Tue Apr 27 19:24:25 2004
@@ -1200,3 +1200,9 @@
 
   quit (exit_status);
 }
+/* AROS */
+/* This function is not in the AROS libc. Someone told me the value once 
+but it's been forgotten. I'm taking a guess here. */
+int getpagesize(void) {
+return 8192;
+}
diff -u -r fileutils-4.1/src/ls.c fileutils-4.1-aros/src/ls.c
--- fileutils-4.1/src/ls.c	Sun Apr 29 05:42:47 2001
+++ fileutils-4.1-aros/src/ls.c	Tue Apr 27 19:24:25 2004
@@ -1047,7 +1047,7 @@
 		 quotearg (p));
 	}
     }
-
+#ifndef __AROS__
 #ifdef TIOCGWINSZ
   {
     struct winsize ws;
@@ -1055,6 +1055,7 @@
     if (ioctl (STDOUT_FILENO, TIOCGWINSZ, &ws) != -1 && ws.ws_col != 0)
       line_length = ws.ws_col;
   }
+#endif
 #endif
 
   /* Using the TABSIZE environment variable is not POSIX-approved.
diff -u -r fileutils-4.1/src/mkfifo.c fileutils-4.1-aros/src/mkfifo.c
--- fileutils-4.1/src/mkfifo.c	Sat Sep 30 05:40:35 2000
+++ fileutils-4.1-aros/src/mkfifo.c	Tue Apr 27 19:24:25 2004
@@ -145,3 +145,13 @@
   exit (errors);
 #endif
 }
+/* AROS */
+
+/* This is not in the aros libc, as aros has no special device files 
+atm. For the purposes of this program, we don't need it and won't use it 
+or care about the results (well hosted ppl might), therefore, return an 
+error. */
+
+int mknod(const char *path, mode_t mode, dev_t dev) {
+return -1;
+}
diff -u -r fileutils-4.1/src/mknod.c fileutils-4.1-aros/src/mknod.c
--- fileutils-4.1/src/mknod.c	Sat Sep 30 04:53:23 2000
+++ fileutils-4.1-aros/src/mknod.c	Tue Apr 27 19:24:25 2004
@@ -236,3 +236,13 @@
 
   exit (0);
 }
+/* AROS */
+
+/* This is not in the aros libc, as aros has no special device files 
+atm. For the purposes of this program, we don't need it and won't use it 
+or care about the results (well hosted ppl might), therefore, return an 
+error. */
+
+int mknod(const char *path, mode_t mode, dev_t dev) {
+return -1;
+}
diff -u -r fileutils-4.1/src/mv.c fileutils-4.1-aros/src/mv.c
--- fileutils-4.1/src/mv.c	Sat Feb  3 11:48:34 2001
+++ fileutils-4.1-aros/src/mv.c	Tue Apr 27 19:24:25 2004
@@ -493,3 +493,19 @@
 
   exit (errors);
 }
+/* AROS */
+
+/* This is not in the aros libc, as aros has no special device files 
+atm. For the purposes of this program, we don't need it and won't use it 
+or care about the results (well hosted ppl might), therefore, return an 
+error. */
+
+int mknod(const char *path, mode_t mode, dev_t dev) {
+return -1;
+}
+
+/* This is not in AROS libc. It could copy the files, but I want to 
+test, so I'll just return not supported or an error. */
+int symlink(const char *name1, const char *name2) {
+return -1;
+}
