diff -ruN fileutils-4.1-aros/config.sub fileutils-4.1/config.sub
--- fileutils-4.1-aros/config.sub	2001-04-21 20:27:49.000000000 +0200
+++ fileutils-4.1/config.sub	2004-11-25 00:02:29.825228344 +0100
@@ -326,7 +326,11 @@
 		basic_machine=m68k-unknown
 		os=-sysv4
 		;;
-	apollo68)
+	aros)
+		basic_machine=i386-unknown
+		os=-aros
+		;;
+		apollo68)
 		basic_machine=m68k-apollo
 		os=-sysv
 		;;
@@ -1009,7 +1013,7 @@
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -solaris* | -sym* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* \
+	      | -aos* | -aros* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -netbsd* | -openbsd* | -freebsd* | -riscix* \
diff -ruN fileutils-4.1-aros/configure fileutils-4.1/configure
--- fileutils-4.1-aros/configure	2001-04-29 10:26:09.000000000 +0200
+++ fileutils-4.1/configure	2004-11-25 00:02:29.836226672 +0100
@@ -17797,7 +17797,7 @@
 if test "${ac_cv_func_setvbuf_reversed+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = foo; then
   { { echo "$as_me:17801: error: cannot run test program while cross compiling" >&5
 echo "$as_me: error: cannot run test program while cross compiling" >&2;}
    { (exit 1); exit 1; }; }
diff -ruN fileutils-4.1-aros/lib/getloadavg.c fileutils-4.1/lib/getloadavg.c
--- fileutils-4.1-aros/lib/getloadavg.c	2000-06-19 09:22:00.000000000 +0200
+++ fileutils-4.1/lib/getloadavg.c	2004-11-25 00:02:29.837226520 +0100
@@ -70,7 +70,7 @@
 
    We also #define LDAV_PRIVILEGED if a program will require
    special installation to be able to call getloadavg.  */
-
+#ifndef __AROS__
 /* This should always be first.  */
 #ifdef HAVE_CONFIG_H
 # include <config.h>
@@ -1084,3 +1084,4 @@
   exit (0);
 }
 #endif /* TEST */
+#endif /* __AROS__ */
diff -ruN fileutils-4.1-aros/lib/getpass.c fileutils-4.1/lib/getpass.c
--- fileutils-4.1-aros/lib/getpass.c	2000-06-21 22:21:03.000000000 +0200
+++ fileutils-4.1/lib/getpass.c	2004-11-25 00:02:29.841225912 +0100
@@ -16,6 +16,7 @@
    write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
    Boston, MA 02111-1307, USA.  */
 
+#ifndef __AROS__
 #if HAVE_CONFIG_H
 # include <config.h>
 #endif
@@ -99,3 +100,4 @@
 
   return buf;
 }
+#endif
diff -ruN fileutils-4.1-aros/lib/mbswidth.c fileutils-4.1/lib/mbswidth.c
--- fileutils-4.1-aros/lib/mbswidth.c	2001-02-17 09:59:57.000000000 +0100
+++ fileutils-4.1/lib/mbswidth.c	2004-11-24 23:59:55.000000000 +0100
@@ -78,6 +78,10 @@
 
 #include "mbswidth.h"
 
+#ifdef __AROS__
+# undef HAVE_MBRTOWC
+#endif
+
 /* Returns the number of columns needed to represent the multibyte
    character string pointed to by STRING.  If a non-printable character
    occurs, -1 is returned, unless MBSW_ACCEPT_UNPRINTABLE is specified.
diff -ruN fileutils-4.1-aros/lib/quotearg.c fileutils-4.1/lib/quotearg.c
--- fileutils-4.1-aros/lib/quotearg.c	2001-02-17 11:39:35.000000000 +0100
+++ fileutils-4.1/lib/quotearg.c	2004-11-24 23:58:21.000000000 +0100
@@ -66,6 +66,10 @@
 # include <wchar.h>
 #endif
 
+#ifdef __AROS__
+# undef HAVE_MBRTOWC
+#endif
+
 #if !HAVE_MBRTOWC
 /* Disable multibyte processing entirely.  Since MB_CUR_MAX is 1, the
    other macros are defined only for documentation and to satisfy C
diff -ruN fileutils-4.1-aros/src/Makefile.in fileutils-4.1/src/Makefile.in
--- fileutils-4.1-aros/src/Makefile.in	2001-04-29 13:40:37.000000000 +0200
+++ fileutils-4.1/src/Makefile.in	2004-11-25 00:02:29.838226368 +0100
@@ -122,9 +122,16 @@
 EXTRA_PROGRAMS = df
 
 bin_PROGRAMS = chgrp chown chmod cp dd dircolors du \
-ginstall ln dir vdir ls mkdir \
-mkfifo mknod mv rm rmdir shred sync touch @DF_PROG@
+dir vdir ls mkdir \
+mkfifo mknod mv rm rmdir touch @DF_PROG@
 
+# AROS - I pulled the following programs from the above list:
+#  ginstall - exec/fork stuff is needed.
+#  shred - Not sure what this is, not a high priority, needs to be 
+#	   looked at
+#  sync - Not looked at all. Not critical atm.
+#  ln - Useless without addressing link() and symlink() maybe fake the 
+#	functions by copying files?
 
 datadir = $(prefix)/@DATADIRNAME@
 localedir = $(datadir)/locale
@@ -174,10 +181,16 @@
 CONFIG_CLEAN_FILES =
 EXTRA_PROGRAMS = df$(EXEEXT)
 bin_PROGRAMS = chgrp$(EXEEXT) chown$(EXEEXT) chmod$(EXEEXT) cp$(EXEEXT) \
-dd$(EXEEXT) dircolors$(EXEEXT) du$(EXEEXT) ginstall$(EXEEXT) \
-ln$(EXEEXT) dir$(EXEEXT) vdir$(EXEEXT) ls$(EXEEXT) mkdir$(EXEEXT) \
+dd$(EXEEXT) dircolors$(EXEEXT) du$(EXEEXT)  \
+dir$(EXEEXT) vdir$(EXEEXT) ls$(EXEEXT) mkdir$(EXEEXT) \
 mkfifo$(EXEEXT) mknod$(EXEEXT) mv$(EXEEXT) rm$(EXEEXT) rmdir$(EXEEXT) \
-shred$(EXEEXT) sync$(EXEEXT) touch$(EXEEXT) @DF_PROG@
+touch$(EXEEXT) @DF_PROG@
+
+# AROS - Removed from above list:
+# ginstall$(EXEEXT)
+# ln$(EXEEXT)
+# shred$(EXEEXT) sync$(EXEEXT)  
+
 PROGRAMS = $(bin_PROGRAMS)
 
 CPPFLAGS = @CPPFLAGS@
diff -ruN fileutils-4.1-aros/src/dd.c fileutils-4.1/src/dd.c
--- fileutils-4.1-aros/src/dd.c	2001-04-22 15:21:58.000000000 +0200
+++ fileutils-4.1/src/dd.c	2004-11-25 00:02:29.839226216 +0100
@@ -1200,3 +1200,9 @@
 
   quit (exit_status);
 }
+/* AROS */
+/* This function is not in the AROS libc. Someone told me the value once 
+but it's been forgotten. I'm taking a guess here. */
+int getpagesize(void) {
+return 8192;
+}
diff -ruN fileutils-4.1-aros/src/ls.c fileutils-4.1/src/ls.c
--- fileutils-4.1-aros/src/ls.c	2001-04-29 11:42:47.000000000 +0200
+++ fileutils-4.1/src/ls.c	2004-11-25 00:02:29.840226064 +0100
@@ -200,6 +200,10 @@
 # define TIMESPEC_NS(timespec) 0
 #endif
 
+#ifdef __AROS__
+# undef HAVE_MBRTOWC
+#endif
+
 enum filetype
   {
     unknown DT_INIT (DT_UNKNOWN),
@@ -1047,7 +1051,7 @@
 		 quotearg (p));
 	}
     }
-
+#ifndef __AROS__
 #ifdef TIOCGWINSZ
   {
     struct winsize ws;
@@ -1056,6 +1060,7 @@
       line_length = ws.ws_col;
   }
 #endif
+#endif
 
   /* Using the TABSIZE environment variable is not POSIX-approved.
      Ignore it when POSIXLY_CORRECT is set.  */
