diff -urN flex-2.5.4/main.c flex-2.5.4-aros/main.c
--- flex-2.5.4/main.c	1997-07-27 04:32:19.000000000 +0200
+++ flex-2.5.4-aros/main.c	2004-09-08 22:07:34.000000000 +0200
@@ -35,6 +35,7 @@
 /* $Header: /home/daffy/u0/vern/flex/RCS/main.c,v 2.64 96/05/25 20:42:42 vern Exp $ */
 
 
+#include <stdlib.h> /* for exit() */
 #include "flexdef.h"
 #include "version.h"
 
diff -urN flex-2.5.4/misc.c flex-2.5.4-aros/misc.c
--- flex-2.5.4/misc.c	1997-07-27 04:32:19.000000000 +0200
+++ flex-2.5.4-aros/misc.c	2004-09-08 22:08:25.000000000 +0200
@@ -28,6 +28,7 @@
 
 /* $Header: /home/daffy/u0/vern/flex/RCS/misc.c,v 2.47 95/04/28 11:39:39 vern Exp $ */
 
+#include <stdlib.h> /* for exit() */
 #include "flexdef.h"
 
 
