diff -ur libtheora-1.1.1.orig/config.sub libtheora-1.1.1/config.sub
--- libtheora-1.1.1.orig/config.sub	2011-12-21 03:34:20.000000000 -0500
+++ libtheora-1.1.1/config.sub	2009-10-01 14:04:08.000000000 -0400
@@ -1216,6 +1216,8 @@
 		;;
 	-none)
 		;;
+	-aros)
+		;;
 	*)
 		# Get rid of the `-' at the beginning of $os.
 		os=`echo $os | sed 's/[^-]*-//'`
diff -ur libtheora-1.1.1.orig/configure libtheora-1.1.1/configure
--- libtheora-1.1.1.orig/configure	2011-12-21 03:30:03.000000000 -0500
+++ libtheora-1.1.1/configure	2011-12-21 03:35:26.000000000 -0500
@@ -11849,7 +11849,6 @@
 fi
 
 
-echo "==== I AM $host ====="
 cflags_save="$CFLAGS"
 if test -z "$GCC"; then
         case $host in
@@ -11861,6 +11860,10 @@
 else
 
         case $host in
+        m68k-*-aros)
+                DEBUG="-g -Wall -Wno-parentheses -DDEBUG -D__NO_MATH_INLINES"
+                CFLAGS="-Wall -Wno-parentheses"
+                PROFILE="-Wall -Wno-parentheses -pg -g -DDEBUG";;
         *)
                 DEBUG="-g -Wall -Wno-parentheses -DDEBUG -D__NO_MATH_INLINES"
                 CFLAGS="-Wall -Wno-parentheses -O3 -fforce-addr -fomit-frame-pointer -finline-functions -funroll-loops"
diff -ur libtheora-1.1.1.orig/configure.ac libtheora-1.1.1/configure.ac
--- libtheora-1.1.1.orig/configure.ac	2009-10-01 14:04:08.000000000 -0400
+++ libtheora-1.1.1/configure.ac	2011-12-21 03:35:15.000000000 -0500
@@ -166,6 +166,10 @@
 else
 
         case $host in 
+        m68k-*-aros)
+                DEBUG="-g -Wall -Wno-parentheses -DDEBUG -D__NO_MATH_INLINES"
+                CFLAGS="-Wall -Wno-parentheses"
+                PROFILE="-Wall -Wno-parentheses -pg -g -DDEBUG";;
         *)
                 DEBUG="-g -Wall -Wno-parentheses -DDEBUG -D__NO_MATH_INLINES"
                 CFLAGS="-Wall -Wno-parentheses -O3 -fforce-addr -fomit-frame-pointer -finline-functions -funroll-loops"
