diff -Naur libtheora-1.1.1-orig/config.sub libtheora-1.1.1/config.sub
--- libtheora-1.1.1-orig/config.sub	2009-02-24 04:50:50.000000000 +0100
+++ libtheora-1.1.1/config.sub	2011-12-27 11:21:52.575097366 +0100
@@ -1216,6 +1216,8 @@
 		;;
 	-none)
 		;;
+	-aros)
+		;;
 	*)
 		# Get rid of the `-' at the beginning of $os.
 		os=`echo $os | sed 's/[^-]*-//'`
diff -Naur libtheora-1.1.1-orig/configure libtheora-1.1.1/configure
--- libtheora-1.1.1-orig/configure	2009-10-01 20:04:08.000000000 +0200
+++ libtheora-1.1.1/configure	2011-12-27 11:22:31.356545090 +0100
@@ -11861,6 +11861,10 @@
 else
 
         case $host in
+        m68k-*-aros)
+                DEBUG="-g -Wall -Wno-parentheses -DDEBUG -D__NO_MATH_INLINES"
+                CFLAGS="-Wall -Wno-parentheses"
+                PROFILE="-Wall -Wno-parentheses -pg -g -DDEBUG";;
         *)
                 DEBUG="-g -Wall -Wno-parentheses -DDEBUG -D__NO_MATH_INLINES"
                 CFLAGS="-Wall -Wno-parentheses -O3 -fforce-addr -fomit-frame-pointer -finline-functions -funroll-loops"
diff -Naur libtheora-1.1.1-orig/configure.ac libtheora-1.1.1/configure.ac
--- libtheora-1.1.1-orig/configure.ac	2009-10-01 20:00:33.000000000 +0200
+++ libtheora-1.1.1/configure.ac	2011-12-27 11:23:23.475691517 +0100
@@ -166,6 +166,10 @@
 else
 
         case $host in 
+        m68k-*-aros)
+                DEBUG="-g -Wall -Wno-parentheses -DDEBUG -D__NO_MATH_INLINES"
+                CFLAGS="-Wall -Wno-parentheses"
+                PROFILE="-Wall -Wno-parentheses -pg -g -DDEBUG";;
         *)
                 DEBUG="-g -Wall -Wno-parentheses -DDEBUG -D__NO_MATH_INLINES"
                 CFLAGS="-Wall -Wno-parentheses -O3 -fforce-addr -fomit-frame-pointer -finline-functions -funroll-loops"
