diff -Naur bwbasic-3.10-orig/bwbasic.h bwbasic-3.10/bwbasic.h
--- bwbasic-3.10-orig/bwbasic.h	2016-12-22 16:49:58.903447206 +0100
+++ bwbasic-3.10/bwbasic.h	2016-12-22 16:56:53.309293335 +0100
@@ -71,6 +71,12 @@
 #endif
 #include         <time.h>
 
+#ifdef __AROS__
+#include <string.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#endif
+
 #ifndef isnan
 #define isnan(x) (x != x)
 #endif
