diff -Naur serialecho-50.5-source-orig/serialecho.c serialecho-50.5-source/serialecho.c
--- serialecho-50.5-source-orig/serialecho.c	2010-07-25 10:44:39.952432666 +0200
+++ serialecho-50.5-source/serialecho.c	2010-07-25 10:45:33.922553047 +0200
@@ -52,6 +52,8 @@
 
 	#define ENTRYPOINT ULONG serialecho(struct ExecBase *SysBase)
 
+	ULONG serialecho(struct ExecBase *SysBase);
+
 	AROS_UFH3(__startup static int, Start,
 	  AROS_UFHA(char *, argstr, A0),
 	  AROS_UFHA(ULONG, argsize, D0),
