diff -Naur sdlscav-145-orig/scav.c sdlscav-145/scav.c
--- sdlscav-145-orig/scav.c	2010-06-28 20:11:22.575317381 +0200
+++ sdlscav-145/scav.c	2010-06-28 20:35:08.472107679 +0200
@@ -1539,7 +1539,11 @@
    strcpy(libpath,LIBPATH);
    strcpy(localdirname,LOCALDIRNAME);
 
+#if defined(__AROS__)
+   p="/PROGDIR";
+#else
    p=getenv(ENVHOME);
+#endif
    if(!p)
    {
       eprintf("Cannot find %s environment variable!\n",ENVHOME);
