diff -Naur sdlscav-145-orig/scav.c sdlscav-145/scav.c
--- sdlscav-145-orig/scav.c	2016-12-28 13:36:35.873846623 +0100
+++ sdlscav-145/scav.c	2016-12-28 13:34:34.918986540 +0100
@@ -1545,7 +1545,11 @@
    strcpy(libpath,LIBPATH);
    strcpy(localdirname,LOCALDIRNAME);
 
+#ifdef __AROS__
+   p="/PROGDIR";
+#else
    p=getenv(ENVHOME);
+#endif
    if(!p)
    {
       eprintf("Cannot find %s environment variable!\n",ENVHOME);
