diff -Naur binutils-2.19/bfd/config.bfd binutils-2.19-aros/bfd/config.bfd
--- binutils-2.19/bfd/config.bfd	2008-04-16 20:02:01.000000000 +0200
+++ binutils-2.19-aros/bfd/config.bfd	2009-09-16 16:46:20.000000000 +0100
@@ -620,6 +620,11 @@
     want64=true
     targ_underscore=yes
     ;;
+  x86_64-*-aros*)
+    targ_defvec=bfd_elf64_x86_64_vec
+    targ_selvecs=bfd_elf32_i386_vec
+    want64=true
+    ;;
 #endif
   i[3-7]86-*-lynxos*)
     targ_defvec=bfd_elf32_i386_vec
@@ -1155,6 +1160,9 @@
     targ_defvec=bfd_powerpcle_pe_vec
     targ_selvecs="bfd_powerpcle_pei_vec bfd_powerpc_pei_vec bfd_powerpcle_pe_vec bfd_powerpc_pe_vec"
     ;;
+  powerpc-*-aros*)
+    targ_defvec=bfd_elf32_powerpc_vec
+    ;;
 
   s390-*-linux*)
     targ_defvec=bfd_elf32_s390_vec
diff -Naur binutils-2.19/ld/configure.tgt binutils-2.19-aros/ld/configure.tgt
--- binutils-2.19/ld/configure.tgt	2008-08-09 07:35:13.000000000 +0200
+++ binutils-2.19-aros/ld/configure.tgt	2009-09-16 16:55:05.000000000 +0100
@@ -239,6 +239,9 @@
 x86_64-*-elf*)		targ_emul=elf_x86_64
 			targ_extra_emuls=elf_i386
 			;;
+x86_64-*-aros*)		targ_emul=elf_x86_64
+			targ_extra_emuls=elf_i386
+			;;
 i[3-7]86-*-kaos*)	targ_emul=elf_i386 ;;
 i[3-7]86-*-freebsdaout* | i[3-7]86-*-freebsd[12].* | i[3-7]86-*-freebsd[12])
 			targ_emul=i386bsd ;;
@@ -487,6 +490,7 @@
 powerpc-*-beos*)	targ_emul=aixppc ;;
 powerpc-*-windiss*)	targ_emul=elf32ppcwindiss ;;
 powerpc-*-lynxos*)	targ_emul=ppclynx ;;
+powerpc-*-aros*)	targ_emul=elf32ppc ;;
 rs6000-*-aix5*)		targ_emul=aix5rs6 ;;
 rs6000-*-aix*)		targ_emul=aixrs6
 			;;
