diff -ru sed-4.1.5/sed/compile.c sed-4.1.5.new/sed/compile.c
--- sed-4.1.5/sed/compile.c	2006-02-03 10:06:26.000000000 +0100
+++ sed-4.1.5.new/sed/compile.c	2008-05-10 22:56:30.000000000 +0200
@@ -1586,7 +1586,7 @@
 
   prog.file = stdin;
   if (cmdfile[0] != '-' || cmdfile[1] != '\0')
-    prog.file = ck_fopen(cmdfile, "rt", true);
+    prog.file = ck_fopen(cmdfile, "r", true);
 
   cur_input.line = 1;
   cur_input.name = cmdfile;
