diff -ruN vice-2.3/configure vice-2.3.aros/configure
--- vice-2.3/configure	2011-02-26 18:07:06.000000000 +0000
+++ vice-2.3.aros/configure	2013-06-20 02:16:50.296361322 +0100
@@ -3380,9 +3380,6 @@
       done
       ;;
   esac
-  if test x"$argvalid" = "xno"; then
-    as_fn_error $? "invalid option: $argcheck" "$LINENO" 5
-  fi
 done
 
 
diff -ruN vice-2.3/src/arch/amigaos/console.c vice-2.3.aros/src/arch/amigaos/console.c
--- vice-2.3/src/arch/amigaos/console.c	2009-08-19 07:12:59.000000000 +0100
+++ vice-2.3.aros/src/arch/amigaos/console.c	2013-06-20 02:27:40.158904200 +0100
@@ -52,7 +52,7 @@
 static BPTR console_handle;
 static int fullscreenwasenabled=0;
 
-#ifdef AMIGA_MORPHOS
+#if defined(AMIGA_MORPHOS) || defined(AMIGA_AROS)
 static APTR fh_putchproc(APTR putchdata, UBYTE ch)
 {
     FPutC((BPTR)putchdata, ch);
@@ -65,12 +65,8 @@
     va_list ap;
 
     va_start(ap, format);
-#ifndef AMIGA_MORPHOS
-#ifdef AMIGA_AROS
-    VFPrintf(console_handle, format, (IPTR *)ap);
-#else
+#if !defined(AMIGA_MORPHOS) && !defined(AMIGA_AROS)
     VFPrintf(console_handle, format, (CONST APTR)ap);
-#endif
 #else
     VNewRawDoFmt(format, fh_putchproc, (STRPTR)console_handle, ap);
 #endif
diff -ruN vice-2.3/src/arch/amigaos/Makefile.in vice-2.3.aros/src/arch/amigaos/Makefile.in
--- vice-2.3/src/arch/amigaos/Makefile.in	2011-02-26 18:07:35.000000000 +0000
+++ vice-2.3.aros/src/arch/amigaos/Makefile.in	2013-06-20 02:16:50.296361322 +0100
@@ -1814,7 +1814,7 @@
 
 intl.h: $(srcdir)/intl.txt
 	echo "generating intl.h..."
-	$(SHELL) $(srcdir)/genintl_h.sh <$(srcdir)/intl.txt >intl.h
+	$(SHELL) genintl_h.sh <$(srcdir)/intl.txt >intl.h
 
 intl_table.h: $(srcdir)/intl.txt
 	echo "generating intl_table.h..."
diff -ruN vice-2.3/src/arch/amigaos/timer.h vice-2.3.aros/src/arch/amigaos/timer.h
--- vice-2.3/src/arch/amigaos/timer.h	2010-12-31 01:41:42.000000000 +0000
+++ vice-2.3.aros/src/arch/amigaos/timer.h	2013-06-20 02:16:50.296361322 +0100
@@ -31,10 +31,6 @@
 
 #include <proto/timer.h>
 
-#ifdef AMIGA_AROS
-#include <aros/_timeval.h>
-#endif
-
 #ifndef AMIGA_OS4
 
 extern void *timer_init(void);
