diff -Naur findutils-4.2.31-org/configure findutils-4.2.31-aros/configure
--- findutils-4.2.31-org/configure	2007-05-30 22:06:17.000000000 +0200
+++ findutils-4.2.31-aros/configure	2009-07-10 08:25:22.000000000 +0200
@@ -1847,13 +1847,9 @@
 gl_func_list="$gl_func_list resolvepath"
 gl_header_list="$gl_header_list fcntl.h"
 gl_func_list="$gl_func_list mempcpy"
-gl_func_list="$gl_func_list btowc"
-gl_func_list="$gl_func_list iswctype"
-gl_func_list="$gl_func_list mbsrtowcs"
 gl_func_list="$gl_func_list wmemchr"
 gl_func_list="$gl_func_list wmemcpy"
 gl_func_list="$gl_func_list wmempcpy"
-gl_header_list="$gl_header_list wctype.h"
 gl_header_list="$gl_header_list stdio_ext.h"
 gl_header_list="$gl_header_list stdint.h"
 gl_header_list="$gl_header_list inttypes.h"
@@ -1864,15 +1860,11 @@
 gl_func_list="$gl_func_list fdopendir"
 gl_func_list="$gl_func_list mbsinit"
 gl_header_list="$gl_header_list locale.h"
-gl_func_list="$gl_func_list mbrtowc"
-gl_func_list="$gl_func_list wcrtomb"
-gl_func_list="$gl_func_list wcscoll"
 gl_func_list="$gl_func_list fchdir"
 gl_header_list="$gl_header_list stdlib.h"
 gl_func_list="$gl_func_list mblen"
 gl_func_list="$gl_func_list mbrlen"
 gl_func_list="$gl_func_list tzset"
-gl_header_list="$gl_header_list wchar.h"
 gl_header_list="$gl_header_list string.h"
 gl_func_list="$gl_func_list lstat"
 gl_header_list="$gl_header_list sys/stat.h"
@@ -18260,6 +18252,7 @@
 
 	gl_cv_func_mbrtowc=no
 fi
+gl_cv_func_mbrtowc=no
 
 rm -f core conftest.err conftest.$ac_objext conftest_ipa8_conftest.oo \
       conftest$ac_exeext conftest.$ac_ext
@@ -20215,6 +20208,7 @@
 
 	gt_cv_c_wchar_t=no
 fi
+gt_cv_c_wchar_t=no
 
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
 fi
diff -Naur findutils-4.2.31-org/find/pred.c findutils-4.2.31-aros/find/pred.c
--- findutils-4.2.31-org/find/pred.c	2007-04-22 17:43:17.000000000 +0200
+++ findutils-4.2.31-aros/find/pred.c	2009-07-10 08:26:25.000000000 +0200
@@ -1563,7 +1563,7 @@
       signal (SIGCHLD, SIG_DFL);
     }
 
-  child_pid = fork ();
+  child_pid = vfork ();
   if (child_pid == -1)
     error (1, errno, _("cannot fork"));
   if (child_pid == 0)
diff -Naur findutils-4.2.31-org/xargs/xargs.c findutils-4.2.31-aros/xargs/xargs.c
--- findutils-4.2.31-org/xargs/xargs.c	2007-03-02 01:14:25.000000000 +0100
+++ findutils-4.2.31-aros/xargs/xargs.c	2009-07-06 14:53:05.000000000 +0200
@@ -1018,7 +1018,7 @@
 	print_args (false);
       /* If we run out of processes, wait for a child to return and
          try again.  */
-      while ((child = fork ()) < 0 && errno == EAGAIN && procs_executing)
+      while ((child = vfork ()) < 0 && errno == EAGAIN && procs_executing)
 	wait_for_proc (false);
       switch (child)
 	{
