diff -Naur hurrican-src-svn281-orig/src/Main.cpp hurrican-src-svn281/src/Main.cpp
--- hurrican-src-svn281-orig/src/Main.cpp	2017-01-05 01:18:14.093422469 +0100
+++ hurrican-src-svn281/src/Main.cpp	2017-01-05 02:33:03.301738013 +0100
@@ -657,7 +657,11 @@
 #else // NON-ANDROID:
 #ifdef USE_HOME_DIR
     // Makefile is specifying this is a UNIX machine and we should write saves,settings,etc to $HOME/.hurrican/ dir
+#ifdef __AROS__
+        char *homedir = "/PROGDIR";
+#else
         char *homedir = getenv("HOME");
+#endif
         bool success = false;
         if (homedir) {
             const char *subdir = "/.hurrican";
diff -Naur hurrican-src-svn281-orig/src/SDLPort/SDL_fmod.cpp hurrican-src-svn281/src/SDLPort/SDL_fmod.cpp
--- hurrican-src-svn281-orig/src/SDLPort/SDL_fmod.cpp	2017-01-05 01:18:14.097422438 +0100
+++ hurrican-src-svn281/src/SDLPort/SDL_fmod.cpp	2017-01-05 01:48:53.798837505 +0100
@@ -353,8 +353,13 @@
     if ((inputmode & FSOUND_LOADMEMORY) == FSOUND_LOADMEMORY)
     {
 #if 1
+#ifdef __AROS__
+        SDL_RWops* sdl_rw = SDL_RWFromMem( (const void*)filename, length );
+		chunk = Mix_LoadWAV_RW(sdl_rw, 1);
+#else
         SDL_RWops* sdl_rw = SDL_RWFromConstMem( (const void*)filename, length );
 		chunk = Mix_LoadWAV_RW(sdl_rw, 1);
+#endif
 #else
         chunk = Mix_QuickLoad_WAV( (Uint8*)filename );
 #endif
diff -Naur hurrican-src-svn281-orig/src/SDLPort/SDL_port.h hurrican-src-svn281/src/SDLPort/SDL_port.h
--- hurrican-src-svn281-orig/src/SDLPort/SDL_port.h	2017-01-05 01:18:14.097422438 +0100
+++ hurrican-src-svn281/src/SDLPort/SDL_port.h	2017-01-05 01:45:38.660384539 +0100
@@ -42,11 +42,13 @@
 
 #define CONST       const
 #ifndef __WIN32__
+#ifndef __AROS__
 #define FALSE       false
 #define TRUE        true
 typedef bool        BOOL;
 typedef float       FLOAT;
 typedef int32_t     LONG;
+#endif
 typedef uint32_t    DWORD, UINT;
 typedef char*       LPSTR;
 typedef const char* LPCTSTR;
diff -Naur hurrican-src-svn281-orig/src/SDLPort/texture.cpp hurrican-src-svn281/src/SDLPort/texture.cpp
--- hurrican-src-svn281-orig/src/SDLPort/texture.cpp	2017-01-05 01:18:14.097422438 +0100
+++ hurrican-src-svn281/src/SDLPort/texture.cpp	2017-01-05 01:49:31.478538785 +0100
@@ -364,8 +364,11 @@
         rawSurf = IMG_Load(fullpath.c_str());
     } else              // Load from memory
     {
+#ifdef __AROS__
+        SDL_RWops* sdl_rw = SDL_RWFromMem( (const void*)buf, buf_size );
+#else
         SDL_RWops* sdl_rw = SDL_RWFromConstMem( (const void*)buf, buf_size );
-
+#endif
         if (sdl_rw != NULL)
         {
             rawSurf = IMG_Load_RW( sdl_rw, 1 );
