diff -Naur vba-m-SVN1292-sources-orig/sources/CMakeLists.txt vba-m-SVN1292-sources/sources/CMakeLists.txt
--- vba-m-SVN1292-sources-orig/sources/CMakeLists.txt	2015-04-04 17:31:34.000000000 +0100
+++ vba-m-SVN1292-sources/sources/CMakeLists.txt	2016-04-02 04:13:36.053664364 +0100
@@ -342,6 +342,8 @@
     )
 	IF( WIN32 )
 		INSTALL(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/vbam${CMAKE_EXECUTABLE_SUFFIX} DESTINATION bin)
+	ELSEIF( AROS )
+		INSTALL(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/vbam DESTINATION .)
 	ELSE( WIN32 )
 		INSTALL(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/vbam DESTINATION bin)
 	ENDIF( WIN32 )
@@ -365,9 +367,9 @@
     endif( ENABLE_NLS )
 endif( ENABLE_GTK OR ENABLE_WX )
 INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/vba-over.ini DESTINATION ${DATA_INSTALL_DIR})
-IF(NOT APPLE)
+IF(NOT APPLE AND NOT AROS)
     INSTALL(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/src/gtk/icons/ DESTINATION share/icons/hicolor PATTERN ".svn" EXCLUDE)
-ENDIF(NOT APPLE)
+ENDIF(NOT APPLE AND NOT AROS)
 SET(CPACK_GENERATOR "ZIP")
 SET(CPACK_PACKAGE_VERSION_MAJOR "1")
 SET(CPACK_PACKAGE_VERSION_MINOR "8")
diff -Naur vba-m-SVN1292-sources-orig/sources/fex/fex/Data_Reader.cpp vba-m-SVN1292-sources/sources/fex/fex/Data_Reader.cpp
--- vba-m-SVN1292-sources-orig/sources/fex/fex/Data_Reader.cpp	2015-04-04 01:05:16.000000000 +0100
+++ vba-m-SVN1292-sources/sources/fex/fex/Data_Reader.cpp	2016-04-02 03:59:08.183102191 +0100
@@ -411,7 +411,11 @@
 		return NULL;
 	
 	size_t needed = 0;
+#ifdef __AROS__
+	size_t mmax = strlen( wpath );
+#else
 	size_t mmax = wcslen( wpath );
+#endif
 	if ( mmax <= 0 )
 		return NULL;
 
diff -Naur vba-m-SVN1292-sources-orig/sources/src/common/Patch.cpp vba-m-SVN1292-sources/sources/src/common/Patch.cpp
--- vba-m-SVN1292-sources-orig/sources/src/common/Patch.cpp	2015-04-04 01:05:20.000000000 +0100
+++ vba-m-SVN1292-sources/sources/src/common/Patch.cpp	2016-04-02 03:59:08.183102191 +0100
@@ -11,6 +11,13 @@
 typedef off_t __off64_t; /* off_t is 64 bits on BSD. */
 #define	fseeko64 fseeko
 #define	ftello64 ftello
+#elif defined(__AROS__)
+#include <exec/types.h>
+#ifndef __off64_t
+typedef QUAD __off64_t;
+#endif
+#define	fseeko64 fseeko
+#define	ftello64 ftello
 #else
 typedef off64_t __off64_t;
 #endif /* __APPLE__ || BSD */
diff -Naur vba-m-SVN1292-sources-orig/sources/src/gba/debugger-expr-yacc.cpp vba-m-SVN1292-sources/sources/src/gba/debugger-expr-yacc.cpp
--- vba-m-SVN1292-sources-orig/sources/src/gba/debugger-expr-yacc.cpp	2015-04-04 01:05:20.000000000 +0100
+++ vba-m-SVN1292-sources/sources/src/gba/debugger-expr-yacc.cpp	2016-04-02 03:59:08.183102191 +0100
@@ -126,7 +126,7 @@
   
 #include "../System.h"
 #include "GBA.h"
-#include "../Common/Port.h"
+#include "../common/Port.h"
 
 #include <string>
 #include <map>
diff -Naur vba-m-SVN1292-sources-orig/sources/src/gba/ereader.cpp vba-m-SVN1292-sources/sources/src/gba/ereader.cpp
--- vba-m-SVN1292-sources-orig/sources/src/gba/ereader.cpp	2015-04-04 01:05:20.000000000 +0100
+++ vba-m-SVN1292-sources/sources/src/gba/ereader.cpp	2016-04-02 03:59:08.183102191 +0100
@@ -1,6 +1,6 @@
 #include <stdio.h>
 #include <locale>
-#include <malloc.h>
+#include <stdlib.h>
 #include <string.h>
 
 #include "GBA.h"
diff -Naur vba-m-SVN1292-sources-orig/sources/src/gba/GBALink.cpp vba-m-SVN1292-sources/sources/src/gba/GBALink.cpp
--- vba-m-SVN1292-sources-orig/sources/src/gba/GBALink.cpp	2015-04-04 01:27:54.000000000 +0100
+++ vba-m-SVN1292-sources/sources/src/gba/GBALink.cpp	2016-04-02 03:59:08.183102191 +0100
@@ -3,12 +3,7 @@
 #include <string.h>
 #include <stdio.h>
 
-// malloc.h does not seem to exist on Mac OS 10.7
-#ifdef __APPLE__
 #include <stdlib.h>
-#else
-#include <malloc.h>
-#endif
 
 #ifdef _MSC_VER
 #define snprintf _snprintf
diff -Naur vba-m-SVN1292-sources-orig/sources/src/gba/remote.cpp vba-m-SVN1292-sources/sources/src/gba/remote.cpp
--- vba-m-SVN1292-sources-orig/sources/src/gba/remote.cpp	2015-04-04 01:05:20.000000000 +0100
+++ vba-m-SVN1292-sources/sources/src/gba/remote.cpp	2016-04-02 03:59:08.183102191 +0100
@@ -29,6 +29,10 @@
 # define write _write
 #endif // _WIN32
 
+#ifdef __AROS__
+struct Library *SocketBase;
+#endif
+
 #include "remote.h"
 #include "BreakpointStructures.h"
 #include "GBA.h"
diff -Naur vba-m-SVN1292-sources-orig/sources/src/sdl/expr.cpp vba-m-SVN1292-sources/sources/src/sdl/expr.cpp
--- vba-m-SVN1292-sources-orig/sources/src/sdl/expr.cpp	2015-04-04 01:05:18.000000000 +0100
+++ vba-m-SVN1292-sources/sources/src/sdl/expr.cpp	2016-04-02 03:59:08.183102191 +0100
@@ -90,12 +90,9 @@
 /* Copy the first part of user declarations.  */
 #line 1 "expr.ypp"
 
-namespace std {  
-#include <stdio.h>
-#include <memory.h>
-#include <stdlib.h>
-#include <string.h>
-}
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
 
 using namespace std;
  
diff -Naur vba-m-SVN1292-sources-orig/sources/src/sdl/SDL.cpp vba-m-SVN1292-sources/sources/src/sdl/SDL.cpp
--- vba-m-SVN1292-sources-orig/sources/src/sdl/SDL.cpp	2015-04-04 01:05:18.000000000 +0100
+++ vba-m-SVN1292-sources/sources/src/sdl/SDL.cpp	2016-04-02 03:59:08.183102191 +0100
@@ -507,6 +507,12 @@
         return f;
     }
   }
+#elif defined(__AROS__)
+  fprintf(stdout, "Searching executable directory\n");
+  sprintf(path, "PROGDIR:etc/%s", name);
+  f = fopen(path, "r");
+  if(f != NULL)
+    return f;
 #else // ! _WIN32
   fprintf(stdout, "Searching data directory: %s\n", PKGDATADIR);
   sprintf(path, "%s%c%s", PKGDATADIR, FILE_SEP, name);
