diff -r -u which-2.16/which.c which-2.16-aros/which.c
--- which-2.16/which.c	Thu Sep 11 20:58:49 2003
+++ which-2.16-aros/which.c	Sun Apr 25 13:53:39 2004
@@ -22,6 +22,9 @@
 #include "getopt.h"
 #include "tilde/tilde.h"
 #include "bash.h"
+#ifdef __AROS__
+#include <ctype.h>
+#endif
 
 static const char *progname;
 
diff -urN which-2.16.orig/mmakefile.src which-2.16/mmakefile.src
--- which-2.16.orig/mmakefile.src	1970-01-01 01:00:00.000000000 +0100
+++ which-2.16/mmakefile.src	2004-05-10 12:05:13.000000000 +0200
@@ -0,0 +1,9 @@
+# Copyright © 2004, The AROS Development Team. All rights reserved.
+# $Id$
+
+include ../../make.cfg
+
+#MM- contrib-gnu : contrib-gnu-which
+	
+%build_with_configure mmake=contrib-gnu-which nix=yes prefix=$(GNUDIR)/which
+
