diff -Naur madbomber-0.2.5-orig/madbomber.c madbomber-0.2.5/madbomber.c
--- madbomber-0.2.5-orig/madbomber.c	2016-12-28 09:56:03.551794326 +0100
+++ madbomber-0.2.5/madbomber.c	2016-12-28 09:42:27.420813567 +0100
@@ -2678,7 +2678,8 @@
       exit(1);
     }
 
-  
+  atexit(SDL_Quit);
+
   /* Init SDL Audio: */
   
   if (use_sound == 1)
@@ -2998,7 +2999,7 @@
 		  "%s\n"
 		  "The Simple DirectMedia error that occured was:\n"
 		  "%s\n\n", MUS_TITLE, SDL_GetError());
-	  exit(1);
+	  // exit(1);
 	}
       
      
@@ -3011,7 +3012,7 @@
 		  "%s\n"
 		  "The Simple DirectMedia error that occured was:\n"
 		  "%s\n\n", MUS_TITLE, SDL_GetError());
-	  exit(1);
+	  // exit(1);
 	}
 #endif
       
@@ -3026,7 +3027,7 @@
 		      "%s\n"
 		      "The Simple DirectMedia error that occured was:\n"
 		      "%s\n\n", game_music_names[i], SDL_GetError());
-	      exit(1);
+	      // exit(1);
 	    }
 	}
     }
@@ -4150,8 +4151,9 @@
   char * filename, * home;
   FILE * fi;
   
-  
-#ifndef WIN32
+#if defined(__AROS__)
+  filename = "PROGDIR:.madbomber";
+#elif !defined(WIN32)
   /* Get home directory (from $HOME variable)... if we can't determine it,
      use the current directory ("."): */
 
