diff -Naur gnurobbo-0.65.6-orig/board.c gnurobbo-0.65.6/board.c
--- gnurobbo-0.65.6-orig/board.c	2010-01-16 16:10:33.000000000 +0100
+++ gnurobbo-0.65.6/board.c	2010-12-25 20:30:37.441737364 +0100
@@ -1897,7 +1897,7 @@
 		  if (rcfile.save_frequency == RCFILE_SAVE_ON_CHANGE)
 		    {
 		      save_resource_file (path_resource_file, FALSE);
-#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
 #elif defined(PLATFORM_GP2X)
 		      system ("sync");
 #elif defined(PLATFORM_ZAURUS)
diff -Naur gnurobbo-0.65.6-orig/controls.c gnurobbo-0.65.6/controls.c
--- gnurobbo-0.65.6-orig/controls.c	2010-01-06 17:04:12.000000000 +0100
+++ gnurobbo-0.65.6/controls.c	2010-12-25 20:30:37.441737364 +0100
@@ -563,7 +563,7 @@
 	/* Set some defaults */
 	for (count = 0; count < USER_CONTROLS; count++)
 	{
-		#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+		#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
 			user_controls[count].device = DEVICE_KEYBOARD;
 		#elif defined(PLATFORM_GP2X)
 			user_controls[count].device = DEVICE_JOYSTICK;
@@ -585,7 +585,7 @@
 	 * you don't want to use a control then just set its device to UNDEFINED
 	 * (id and mod are already set to UNDEFINED from the above code) otherwise
 	 * within Options you'll see the device shown on its own which looks weird */
-	#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+	#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
 		user_controls[ACTION_UP].id = SDLK_UP;
 		user_controls[ACTION_UP_RIGHT].device = UNDEFINED;
 		user_controls[ACTION_RIGHT].id = SDLK_RIGHT;
@@ -824,7 +824,7 @@
 	{
 		/* WARNING: attempting to close joystick 0 on the GP2X
 		 * using the GPH SDK causes a seg fault */
-		#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+		#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
 			if (joystick)
 			{
 		#elif defined(PLATFORM_GP2X)
diff -Naur gnurobbo-0.65.6-orig/game.c gnurobbo-0.65.6/game.c
--- gnurobbo-0.65.6-orig/game.c	2010-01-16 16:10:33.000000000 +0100
+++ gnurobbo-0.65.6/game.c	2010-12-25 20:30:37.441737364 +0100
@@ -183,7 +183,7 @@
   pointer_controls.state = POINTER_CONTROLS_MODE_OFF;
   pointer_controls.shoot_state = POINTER_CONTROLS_SHOOT_OFF;
   pointer_controls.pad_type = POINTER_CONTROLS_PAD_TYPE_VIEWPORT;	/* POINTER_CONTROLS_PAD_TYPE_SCREEN; */
-#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
   gnurobbo_op_env.systempointer = TRUE;
 #elif defined(PLATFORM_GP2X)
   gnurobbo_op_env.systempointer = FALSE;
@@ -351,6 +351,8 @@
 	/* Set-up the resource file path for the required platform and read it now */
 	#if defined(PLATFORM_WIN32)
 		strcpy (path_resource_file, getenv("TEMP"));
+	#elif defined(PLATFORM_AROS)
+		strcpy (path_resource_file, "PROGDIR:");
 	#elif defined(PLATFORM_PC)
 		strcpy (path_resource_file, getenv ("HOME"));
 		strcat (path_resource_file, "/");
@@ -443,7 +445,11 @@
   sort_locales ();
 
   /* Initialise SDL */
+#if defined(PLATFORM_AROS)
+  if (SDL_Init (SDL_INIT_TIMER | SDL_INIT_AUDIO | SDL_INIT_VIDEO))
+#else
   if (SDL_Init (SDL_INIT_EVERYTHING))
+#endif
     {
       fprintf (stdout, "Cannot initialise SDL: %s", SDL_GetError ());
       return 1;
@@ -462,7 +468,7 @@
 	  initialise_joystick (UNDEFINED, default_joystick_name, TRUE);
     }
 
-#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
   /* Load the window manager icon and set caption */
   set_wm_icon();
   SDL_WM_SetCaption ("GNU Robbo", "gnurobbo");
@@ -1597,7 +1603,7 @@
 
   /* WARNING: attempting to close joystick 0 on the GP2X using the GPH
    * SDK causes a seg fault and it may do the same on similar devices */
-#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
   if (joystick)
     SDL_JoystickClose (joystick);
 #elif defined(PLATFORM_GP2X)
@@ -2178,7 +2184,7 @@
 	  if (rcfile.save_frequency == RCFILE_SAVE_ON_CHANGE)
 	    {
 	      save_resource_file (path_resource_file, FALSE);
-#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
 #elif defined(PLATFORM_GP2X)
 	      system ("sync");
 #elif defined(PLATFORM_ZAURUS)
diff -Naur gnurobbo-0.65.6-orig/levels.c gnurobbo-0.65.6/levels.c
--- gnurobbo-0.65.6-orig/levels.c	2010-01-16 23:33:19.000000000 +0100
+++ gnurobbo-0.65.6/levels.c	2010-12-25 20:30:37.441737364 +0100
@@ -182,7 +182,7 @@
 	FILE *fp;
 	char fullpath[256];
 	#if defined(PLATFORM_WIN32)
-	#elif defined(PLATFORM_PC) || defined(PLATFORM_ZAURUS)
+	#elif defined(PLATFORM_PC) || defined(PLATFORM_ZAURUS) || defined(PLATFORM_AROS)
 		char foldername[256];
 	#elif defined(PLATFORM_GP2X)
 	#elif defined(PLATFORM_PSP)
@@ -194,6 +194,8 @@
 	#elif defined(PLATFORM_PC) || defined(PLATFORM_ZAURUS)
 		strcpy (fullpath, getenv ("HOME"));
 		strcat (fullpath, "/" LOCAL_DATA_DIR LEVELS_DIR DEFAULT_USER_LEVEL_PACK);
+	#elif defined(PLATFORM_AROS)
+		strcpy (fullpath, "PROGDIR:" LOCAL_DATA_DIR LEVELS_DIR DEFAULT_USER_LEVEL_PACK);
 	#elif defined(PLATFORM_GP2X) || defined(PLATFORM_PSP)
 		strcpy (fullpath, PACKAGE_DATA_DIR LEVELS_DIR DEFAULT_USER_LEVEL_PACK);
 	#endif
@@ -208,6 +210,11 @@
 
 			/* Create directory structure ignoring errors */
 			#if defined(PLATFORM_WIN32)
+			#elif defined(PLATFORM_AROS)
+				strcpy (foldername, "PROGDIR:" LOCAL_DATA_DIR);
+				mkdir (foldername, 0755);
+				strcat (foldername, LEVELS_DIR);
+				mkdir (foldername, 0755);
 			#elif defined(PLATFORM_PC) || defined(PLATFORM_ZAURUS)
 				strcpy (foldername, getenv ("HOME"));
 				strcat (foldername, "/" LOCAL_DATA_DIR);
@@ -299,7 +306,7 @@
 
 	#if defined(PLATFORM_WIN32)
 		for (count = 0; count < 1; count++)
-	#elif defined(PLATFORM_PC) || defined(PLATFORM_ZAURUS)
+	#elif defined(PLATFORM_PC) || defined(PLATFORM_ZAURUS) || defined(PLATFORM_AROS)
 		for (count = 0; count < 2; count++)
 	#elif defined(PLATFORM_GP2X) || defined(PLATFORM_PSP)
 		for (count = 0; count < 1; count++)
@@ -312,6 +319,8 @@
 		else if (count == 1)
 		{
 			#if defined(PLATFORM_WIN32)
+			#elif defined(PLATFORM_AROS)
+				strcpy (foldername, "PROGDIR:" LOCAL_DATA_DIR LEVELS_DIR);
 			#elif defined(PLATFORM_PC) || defined(PLATFORM_ZAURUS)
 				strcpy (foldername, getenv ("HOME"));
 				strcat (foldername, "/" LOCAL_DATA_DIR LEVELS_DIR);
diff -Naur gnurobbo-0.65.6-orig/locales.c gnurobbo-0.65.6/locales.c
--- gnurobbo-0.65.6-orig/locales.c	2010-01-06 00:02:23.000000000 +0100
+++ gnurobbo-0.65.6/locales.c	2010-12-25 20:30:37.441737364 +0100
@@ -324,6 +324,10 @@
 		strcpy (config_items[++row].name, "[txt_intro_keys_PLATFORM_PC]");
 		config_items[row].destination = txt_intro_keys;
 		config_items[row].max_rows = TXT_INTRO_KEYS_ROWS;
+	#elif defined(PLATFORM_AROS)
+		strcpy (config_items[++row].name, "[txt_intro_keys_PLATFORM_AROS]");
+		config_items[row].destination = txt_intro_keys;
+		config_items[row].max_rows = TXT_INTRO_KEYS_ROWS;
 	#elif defined(PLATFORM_GP2X)
 		strcpy (config_items[++row].name, "[txt_intro_keys_PLATFORM_GP2X]");
 		config_items[row].destination = txt_intro_keys;
diff -Naur gnurobbo-0.65.6-orig/rcfile.h gnurobbo-0.65.6/rcfile.h
--- gnurobbo-0.65.6-orig/rcfile.h	2010-01-05 01:18:39.000000000 +0100
+++ gnurobbo-0.65.6/rcfile.h	2010-12-25 20:30:37.441737364 +0100
@@ -23,6 +23,8 @@
 	#define RESOURCE_FILE "gnurobborc"
 #elif defined(PLATFORM_PC)
 	#define RESOURCE_FILE ".gnurobborc"
+#elif defined(PLATFORM_AROS)
+	#define RESOURCE_FILE ".gnurobborc"
 #elif defined(PLATFORM_GP2X)
 	#define RESOURCE_FILE "gnurobborc"
 #elif defined(PLATFORM_ZAURUS)
diff -Naur gnurobbo-0.65.6-orig/screen.c gnurobbo-0.65.6/screen.c
--- gnurobbo-0.65.6-orig/screen.c	2010-01-16 21:41:55.000000000 +0100
+++ gnurobbo-0.65.6/screen.c	2010-12-25 20:30:37.451737205 +0100
@@ -1248,7 +1248,7 @@
 	video.field_size = 16;
     }
 
-#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
 #elif defined(PLATFORM_GP2X)
   /* Force GP2X screen resolution as there is no option. */
   video.xres = 320; video.yres = 240; video.field_size = 16;
@@ -1260,7 +1260,7 @@
   if (video.fullscreen == UNDEFINED)
     video.fullscreen = 0;
 
-#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
 #elif defined(PLATFORM_GP2X)
   video.fullscreen = SDL_FULLSCREEN;
 #elif defined(PLATFORM_ZAURUS)
@@ -1363,7 +1363,7 @@
   printf ("BytesPerPixel: %u\n", videoinfo->vfmt->BytesPerPixel);
 #endif
 
-#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) ||defined(PLATFORM_AROS)
 #elif defined(PLATFORM_GP2X)
   /* Well after all that clever video mode detection code the GPH GP2X
    * SDL states it has hwsurface support and then seg faults ;) */
@@ -3048,7 +3048,7 @@
 		options[(OPTIONS_SCREEN_PAGES - 1) * 13 + OPTIONS_MENUPOS_RIGHT] = FALSE;
 
 		/* Disable Toggle Fullscreen on platforms that don't support this */
-		#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+		#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
 		#elif defined(PLATFORM_GP2X)
 			options[OPTIONS_ACTION_TOGGLE_FULLSCREEN] = FALSE;
 		#elif defined(PLATFORM_ZAURUS)
@@ -3058,7 +3058,7 @@
 		#endif
 
 		/* Disable pointer options on platforms that don't support the mouse */
-		#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC)
+		#if defined(PLATFORM_WIN32) || defined(PLATFORM_PC) || defined(PLATFORM_AROS)
 		#elif defined(PLATFORM_GP2X)
 			#ifndef HAVE_DESIGNER
 				/* F100 doesn't support the mouse, F200 does (or can) so
diff -Naur gnurobbo-0.65.6-orig/sound.c gnurobbo-0.65.6/sound.c
--- gnurobbo-0.65.6-orig/sound.c	2010-01-05 01:18:39.000000000 +0100
+++ gnurobbo-0.65.6/sound.c	2010-12-27 10:08:27.311019891 +0100
@@ -413,11 +413,11 @@
 	sounds[c].initialized = SND_MISSING;
     }
 
-    sprintf(name, "%s%s/skins/%s", PACKAGE_DATA_DIR, SOUNDS_DIR, fname);
+    sprintf(name, "%s%sskins/%s", PACKAGE_DATA_DIR, SOUNDS_DIR, fname);
     if ((fd = fopen(name, "r")) == NULL) {	/* no skin file, well,
 						 * what a pity, let's try
 						 * the default one */
-	sprintf(name, "%s%s/skins/default.dat", PACKAGE_DATA_DIR,
+	sprintf(name, "%s%sskins/default.dat", PACKAGE_DATA_DIR,
 		SOUNDS_DIR);
 	if ((fd = fopen(name, "r")) == NULL)	/* hmmm no default sound * 
 						 * skin, how strange */
