diff -ruN lzo-2.03/src/miniacc.h lzo-2.03.aros/src/miniacc.h
--- lzo-2.03/src/miniacc.h	2008-04-30 14:51:49.000000000 +0200
+++ lzo-2.03.aros/src/miniacc.h	2008-09-30 13:17:16.000000000 +0200
@@ -2417,6 +2417,9 @@
 #  undef HAVE_UMASK
 #  undef HAVE_UTIME
 #endif
+#if defined(__AROS__)
+#  undef HAVE_MALLOC_H
+#endif
 #if (ACC_LIBC_ISOC90 || ACC_LIBC_ISOC99)
 #  undef HAVE_ACCESS
 #  undef HAVE_CHMOD
