diff -ur libxml2-2.7.8/runtest.c libxml2-2.7.8-aros/runtest.c
--- libxml2-2.7.8/runtest.c	2010-10-12 08:25:32.000000000 +0200
+++ libxml2-2.7.8-aros/runtest.c	2011-06-10 20:59:56.000000000 +0200
@@ -29,7 +29,7 @@
 #include <libxml/tree.h>
 #include <libxml/uri.h>
 
-#ifdef LIBXML_OUTPUT_ENABLED
+#if definef LIBXML_OUTPUT_ENABLED && !defined __AROS__
 #ifdef LIBXML_READER_ENABLED
 #include <libxml/xmlreader.h>
 #endif
diff -ur libxml2-2.7.8/testrecurse.c libxml2-2.7.8-aros/testrecurse.c
--- libxml2-2.7.8/testrecurse.c	2010-10-12 08:25:32.000000000 +0200
+++ libxml2-2.7.8-aros/testrecurse.c	2011-06-10 21:11:49.000000000 +0200
@@ -16,6 +16,8 @@
 #include <stdio.h>
 #endif
 
+#if !defined(__AROS__)
+
 #if !defined(_WIN32) || defined(__CYGWIN__)
 #include <unistd.h>
 #endif
@@ -971,3 +973,8 @@
 
     return(ret);
 }
+#else
+int main()
+{
+}
+#endif
