diff -ur libxml2-2.6.31/runtest.c libxml2-2.6.31-aros/runtest.c
--- libxml2-2.6.31/runtest.c	2008-01-11 09:01:56.000000000 +0100
+++ libxml2-2.6.31-aros/runtest.c	2008-08-25 21:22:13.000000000 +0200
@@ -29,7 +29,7 @@
 #include <libxml/tree.h>
 #include <libxml/uri.h>
 
-#ifdef LIBXML_OUTPUT_ENABLED
+#if definef LIBXML_OUTPUT_ENABLED && !defined __AROS__
 #ifdef LIBXML_READER_ENABLED
 #include <libxml/xmlreader.h>
 #endif
