diff -ruN ArxLibertatis-1.1.2-r1/CMakeLists.txt ArxLibertatis-1.1.2-r1.aros/CMakeLists.txt
--- ArxLibertatis-1.1.2-r1/CMakeLists.txt	2016-03-24 22:50:29.000000000 +0100
+++ ArxLibertatis-1.1.2-r1.aros/CMakeLists.txt	2018-08-22 15:35:48.622489341 +0200
@@ -127,7 +127,9 @@
 )
 
 # Default runtime user and data directories
-if(WIN32)
+if(AROS)
+# AROS uses binary dir for data.
+elseif(WIN32)
 	set(USER_DIR            "Arx Libertatis"                CACHE STRING "User dir names")
 	set(USER_DIR_PREFIXES   "%FOLDERID_SavedGames%"         CACHE STRING "User dir paths")
 elseif(MACOSX)
@@ -190,6 +192,7 @@
 include(PrintConfiguration)
 include(StyleCheck)
 include(VersionString)
+include(CheckIncludeFiles)
 
 
 # Find required libraries
diff -ruN ArxLibertatis-1.1.2-r1/src/core/ArxGame.cpp ArxLibertatis-1.1.2-r1.aros/src/core/ArxGame.cpp
--- ArxLibertatis-1.1.2-r1/src/core/ArxGame.cpp	2016-03-24 22:50:29.000000000 +0100
+++ ArxLibertatis-1.1.2-r1.aros/src/core/ArxGame.cpp	2018-08-22 03:39:51.016571901 +0200
@@ -49,6 +49,11 @@
 #include <algorithm>
 #include <sstream>
 
+#ifdef __AROS__
+#include <aros/preprocessor/variadic/size.hpp>
+#include <aros/preprocessor/variadic/elem.hpp>
+#endif
+
 #include <boost/foreach.hpp>
 
 #include "ai/PathFinderManager.h"
diff -ruN ArxLibertatis-1.1.2-r1/src/gui/MenuWidgets.cpp ArxLibertatis-1.1.2-r1.aros/src/gui/MenuWidgets.cpp
--- ArxLibertatis-1.1.2-r1/src/gui/MenuWidgets.cpp	2016-03-24 22:50:29.000000000 +0100
+++ ArxLibertatis-1.1.2-r1.aros/src/gui/MenuWidgets.cpp	2018-08-22 03:39:51.016571901 +0200
@@ -87,7 +87,7 @@
 
 #include "window/RenderWindow.h"
 
-using std::wistringstream;
+//using std::wistringstream;
 using std::min;
 using std::max;
 using std::string;
diff -ruN ArxLibertatis-1.1.2-r1/src/input/Input.cpp ArxLibertatis-1.1.2-r1.aros/src/input/Input.cpp
--- ArxLibertatis-1.1.2-r1/src/input/Input.cpp	2016-03-24 22:50:29.000000000 +0100
+++ ArxLibertatis-1.1.2-r1.aros/src/input/Input.cpp	2018-08-22 03:39:51.016571901 +0200
@@ -46,6 +46,11 @@
 
 #include "input/Input.h"
 
+#ifdef __AROS__
+#include <aros/preprocessor/variadic/size.hpp>
+#include <aros/preprocessor/variadic/elem.hpp>
+#endif
+
 #include <string>
 #include <map>
 
diff -ruN ArxLibertatis-1.1.2-r1/src/io/fs/FilesystemPOSIX.cpp ArxLibertatis-1.1.2-r1.aros/src/io/fs/FilesystemPOSIX.cpp
--- ArxLibertatis-1.1.2-r1/src/io/fs/FilesystemPOSIX.cpp	2016-03-24 22:50:29.000000000 +0100
+++ ArxLibertatis-1.1.2-r1.aros/src/io/fs/FilesystemPOSIX.cpp	2018-08-22 03:39:51.016571901 +0200
@@ -254,7 +254,12 @@
 	do {
 		
 		dirent * entry;
+#ifdef __AROS__
+//FIXME:possibly mutex necessary
+		if(readdir(handle)) {
+#else
 		if(readdir_r(handle, buf, &entry) || !entry) {
+#endif
 			std::free(_buf), _buf = NULL;
 			return;
 		}
