diff -Naur grafx2-orig/share/grafx2/gfx2def.ini grafx2/share/grafx2/gfx2def.ini
--- grafx2-orig/share/grafx2/gfx2def.ini	2012-07-29 13:40:58.612496836 +0200
+++ grafx2/share/grafx2/gfx2def.ini	2012-07-29 09:13:47.392442984 +0200
@@ -327,12 +327,12 @@
   ; Name of the skinfile you want to  | Nom du fichier skin que vous voulez
   ; use.                              | utiliser.
   ; Default : (empty to let the program choose)
-  Skin_file = 
+  Skin_file = skin_classic.png
 
   ; Name of the font file (8x8) you   | Nom du fichier police de caractère
   ; want to use.                      | 8x8 utilisée dans les menus.
   ; Default : (empty to let the program choose)
-  Font_file = 
+  Font_file = font_Classic.png
 
   ; This determines the color value for the grid. Each pixel of 
   ; the grid will be displayed by XOR-ing the original color with
diff -Naur grafx2-orig/src/filesel.c grafx2/src/filesel.c
--- grafx2-orig/src/filesel.c	2012-07-29 13:40:58.612496836 +0200
+++ grafx2/src/filesel.c	2012-07-29 13:38:21.324942842 +0200
@@ -460,13 +460,13 @@
   STRPTR iptr;
   dword i;
 
-  iptr = BADDR( in );
-
-  if( max > iptr[0] ) max = iptr[0];
-
 #if defined(__AROS__)
-  for ( i=0 ; i<max ; i++ ) out[i] = *(AROS_BSTR_ADDR(iptr+i));
+  iptr = AROS_BSTR_ADDR( in );
+  if( max > AROS_BSTR_strlen(in) ) max = AROS_BSTR_strlen(in);
+  for( i=0; i<max; i++ ) out[i] = iptr[i];
 #else
+  iptr = BADDR( in );
+  if( max > iptr[0] ) max = iptr[0];
   for( i=0; i<max; i++ ) out[i] = iptr[i+1];
 #endif
   out[i] = 0;
diff -Naur grafx2-orig/src/setup.c grafx2/src/setup.c
--- grafx2-orig/src/setup.c	2012-07-29 13:40:58.612496836 +0200
+++ grafx2/src/setup.c	2012-07-29 09:13:47.392442984 +0200
@@ -135,7 +135,7 @@
     strcpy(data_dir, program_dir);
   // All other targets, program is in a "bin" subdirectory
   #elif defined (__AROS__)
-    strcat(data_dir,"/share/grafx2/");
+    strcat(data_dir,"share/grafx2/");
   #else
     strcat(data_dir,"../share/grafx2/");
   #endif
