diff -Naur New/ZunePaint/src/Win_GradDef.c ZunePaint_v0.13/ZunePaint/src/Win_GradDef.c
--- New/ZunePaint/src/Win_GradDef.c	2017-03-25 17:20:22.000000000 +0100
+++ ZunePaint_v0.13/ZunePaint/src/Win_GradDef.c	2018-07-21 16:30:08.817818052 +0200
@@ -25,7 +25,7 @@
 #include "Picture_Paint.h"
 #include "Picture_GradientFill.h"
 #include "mcc_GradEdit.h"
-#include "mcc_graddisp.h"
+#include "mcc_Graddisp.h"
 #include "Picture_Color.h"
 #include "locale.h"
 
diff -Naur New/ZunePaint/src/ZunePaint.c ZunePaint_v0.13/ZunePaint/src/ZunePaint.c
--- New/ZunePaint/src/ZunePaint.c	2018-04-15 08:27:44.000000000 +0200
+++ ZunePaint_v0.13/ZunePaint/src/ZunePaint.c	2018-07-21 16:28:36.421855832 +0200
@@ -39,7 +39,7 @@
 #include "SupportFunctions.h"
 #include "mcc_Pictdisp.h"
 #include "mcc_ScrollerPict.h"
-#include "mcc_HistDisp.h"
+#include "mcc_Histdisp.h"
 #include "mcc_Matrix.h"
 #include "locale.h"
 #include "Picture_Geometry.h"
