diff -Naur ZunePaint_v0.11-orig/ZunePaint/src/Picture_Paint.c ZunePaint_v0.11/ZunePaint/src/Picture_Paint.c
--- ZunePaint_v0.11-orig/ZunePaint/src/Picture_Paint.c	2017-07-31 12:31:02.447787084 +0200
+++ ZunePaint_v0.11/ZunePaint/src/Picture_Paint.c	2017-07-31 12:44:15.045208367 +0200
@@ -7,6 +7,7 @@
 **       AROS Public License
 */
 
+#include <proto/arossupport.h>
 #include <proto/exec.h>
 #include <proto/dos.h>
 #include <proto/muimaster.h>
@@ -25,7 +26,7 @@
 #include <math.h>
 #include <string.h>
 
-#include "mcc_PictDisp.h"
+#include "mcc_Pictdisp.h"
 #include "Picture_Masks.h"
 #include "Picture_Paint.h"
 #include "Picture_Color.h"
diff -Naur ZunePaint_v0.11-orig/ZunePaint/src/Win_GradDef.c ZunePaint_v0.11/ZunePaint/src/Win_GradDef.c
--- ZunePaint_v0.11-orig/ZunePaint/src/Win_GradDef.c	2017-07-31 12:31:02.451787049 +0200
+++ ZunePaint_v0.11/ZunePaint/src/Win_GradDef.c	2017-07-31 12:45:57.668383236 +0200
@@ -25,7 +25,7 @@
 #include "Picture_Paint.h"
 #include "Picture_GradientFill.h"
 #include "mcc_GradEdit.h"
-#include "mcc_graddisp.h"
+#include "mcc_Graddisp.h"
 #include "Picture_Color.h"
 #include "locale.h"
 
diff -Naur ZunePaint_v0.11-orig/ZunePaint/src/ZunePaint.c ZunePaint_v0.11/ZunePaint/src/ZunePaint.c
--- ZunePaint_v0.11-orig/ZunePaint/src/ZunePaint.c	2017-07-31 12:31:02.451787049 +0200
+++ ZunePaint_v0.11/ZunePaint/src/ZunePaint.c	2017-07-31 12:47:28.771653214 +0200
@@ -39,7 +39,7 @@
 #include "SupportFunctions.h"
 #include "mcc_Pictdisp.h"
 #include "mcc_ScrollerPict.h"
-#include "mcc_HistDisp.h"
+#include "mcc_Histdisp.h"
 #include "mcc_Matrix.h"
 #include "locale.h"
 #include "Picture_Geometry.h"
