diff -Naur ZuneView_v0.31-orig/ZuneView/src/mcc_Pictdisp.c ZuneView_v0.31/ZuneView/src/mcc_Pictdisp.c
--- ZuneView_v0.31-orig/ZuneView/src/mcc_Pictdisp.c	2016-12-23 23:08:53.576135879 +0100
+++ ZuneView_v0.31/ZuneView/src/mcc_Pictdisp.c	2016-12-23 23:15:48.533164658 +0100
@@ -62,6 +62,26 @@
 void ClearSelection(Object *obj, struct Selection *sel);					/* Clears the selection area information */
 void ColorPickerPointerUpdate(struct Window *win, struct PictDispData *data, ULONG color);
 
+#ifdef AROS_ABI_V1
+struct RastPort *CreateRastPort(void)
+{
+    struct RastPort *rp = AllocMem(sizeof (struct RastPort), MEMF_ANY);
+    if (rp)
+    {
+        InitRastPort(rp);
+    }
+    return rp;
+}
+
+void FreeRastPort(struct RastPort *rp)
+{
+    if (rp)
+    {
+        FreeMem(rp, sizeof (struct RastPort));
+    }
+}
+#endif
+
 /* Global static variable */
 UBYTE  checkerboard[32] =
 {
