diff -ruN curl-7.60.0/acinclude.m4 curl-7.60.0.aros/acinclude.m4
--- curl-7.60.0/acinclude.m4	2018-05-07 10:18:02.000000000 +0100
+++ curl-7.60.0.aros/acinclude.m4	2018-06-09 06:45:42.615317739 +0100
@@ -1342,6 +1342,12 @@
 #ifdef HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #endif
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+#endif
+#ifdef __AROS__
+      struct Library *SocketBase;
 #endif
     ]],[[
       recv(0, 0, 0, 0);
@@ -1479,6 +1485,12 @@
 #ifdef HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #endif
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+#endif
+#ifdef __AROS__
+      struct Library *SocketBase;
 #endif
     ]],[[
       send(0, 0, 0, 0);
@@ -2047,6 +2059,15 @@
     fi
   done
   LIBS="$tst_connect_save_LIBS"
+  if test "$tst_connect_need_LIBS" = unknown; then
+      dnl on AROS, connect is in bsdsocket.library
+      OLIBS=$LIBS
+      LIBS="-lssl $LIBS"
+      AC_TRY_LINK([#include <proto/bsdsocket.h>],
+                  [void *SocketBase; connect(0,0,0);],
+                  [tst_connect_need_LIBS=""],
+                  [])
+  fi
   #
   case X-"$tst_connect_need_LIBS" in
     X-unknown)
diff -ruN curl-7.60.0/configure curl-7.60.0.aros/configure
--- curl-7.60.0/configure	2018-05-14 13:20:30.000000000 +0100
+++ curl-7.60.0.aros/configure	2018-06-09 06:45:42.615317739 +0100
@@ -1097,7 +1097,6 @@
 docdir
 oldincludedir
 includedir
-runstatedir
 localstatedir
 sharedstatedir
 sysconfdir
@@ -1253,7 +1252,6 @@
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
-runstatedir='${localstatedir}/run'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
@@ -1506,15 +1504,6 @@
   | -silent | --silent | --silen | --sile | --sil)
     silent=yes ;;
 
-  -runstatedir | --runstatedir | --runstatedi | --runstated \
-  | --runstate | --runstat | --runsta | --runst | --runs \
-  | --run | --ru | --r)
-    ac_prev=runstatedir ;;
-  -runstatedir=* | --runstatedir=* | --runstatedi=* | --runstated=* \
-  | --runstate=* | --runstat=* | --runsta=* | --runst=* | --runs=* \
-  | --run=* | --ru=* | --r=*)
-    runstatedir=$ac_optarg ;;
-
   -sbindir | --sbindir | --sbindi | --sbind | --sbin | --sbi | --sb)
     ac_prev=sbindir ;;
   -sbindir=* | --sbindir=* | --sbindi=* | --sbind=* | --sbin=* \
@@ -1652,7 +1641,7 @@
 for ac_var in	exec_prefix prefix bindir sbindir libexecdir datarootdir \
 		datadir sysconfdir sharedstatedir localstatedir includedir \
 		oldincludedir docdir infodir htmldir dvidir pdfdir psdir \
-		libdir localedir mandir runstatedir
+		libdir localedir mandir
 do
   eval ac_val=\$$ac_var
   # Remove trailing slashes.
@@ -1805,7 +1794,6 @@
   --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
-  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
   --libdir=DIR            object code libraries [EPREFIX/lib]
   --includedir=DIR        C header files [PREFIX/include]
   --oldincludedir=DIR     C header files for non-gcc [/usr/include]
@@ -7392,7 +7380,7 @@
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -7437,7 +7425,7 @@
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -7460,7 +7448,7 @@
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -7504,7 +7492,7 @@
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -7527,7 +7515,7 @@
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -19676,6 +19664,33 @@
 fi
 
 
+if test "$HAVE_GETHOSTBYNAME" != "1"
+then
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for gethostbyname in bsdsocket.library" >&5
+$as_echo_n "checking for gethostbyname in bsdsocket.library... " >&6; }
+  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+#include <proto/bsdsocket.h>
+int main (void)
+{
+void *SocketBase; gethostbyname("");
+ ;
+ return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"; then :
+                 { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
+$as_echo "yes" >&6; }
+              HAVE_GETHOSTBYNAME="1"
+else
+  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
+$as_echo "no" >&6; }
+
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+fi
+
 
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether to provide built-in manual" >&5
@@ -20409,6 +20424,25 @@
     fi
   done
   LIBS="$tst_connect_save_LIBS"
+  if test "$tst_connect_need_LIBS" = unknown; then
+            OLIBS=$LIBS
+      LIBS="-lssl $LIBS"
+      cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+#include <proto/bsdsocket.h>
+int main (void)
+{
+void *SocketBase; connect(0,0,0);
+ ;
+ return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"; then :
+  tst_connect_need_LIBS=""
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+  fi
   #
   case X-"$tst_connect_need_LIBS" in
     X-unknown)
@@ -22893,8 +22927,38 @@
             { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
         fi
+    fi
 
-    else
+    if test "$ac_cv_lib_ssl_SSL_connect" != yes; then
+                { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ssl with SocketBase available" >&5
+$as_echo_n "checking for ssl with SocketBase available... " >&6; }
+        OLIBS=$LIBS
+        LIBS="-lssl $LIBS"
+        cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+#include <openssl/ssl.h>
+void *SocketBase;
+int main (void)
+{
+SSL_connect(0);
+ ;
+ return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"; then :
+  ac_cv_lib_ssl_SSL_connect="yes"
+                     { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
+$as_echo "yes" >&6; }
+else
+  LIBS=$OLIBS
+                     { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
+$as_echo "no" >&6; }
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+    fi
+
+    if test "$ac_cv_lib_ssl_SSL_connect" = yes; then
 
             for ac_header in openssl/x509.h openssl/rsa.h openssl/crypto.h \
                        openssl/pem.h openssl/ssl.h openssl/err.h
@@ -30031,6 +30095,12 @@
 #ifdef HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #endif
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+#endif
+#ifdef __AROS__
+      struct Library *SocketBase;
 #endif
 
 int main (void)
@@ -30222,6 +30292,12 @@
 #ifdef HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #endif
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+#endif
+#ifdef __AROS__
+      struct Library *SocketBase;
 #endif
 
 int main (void)
diff -ruN curl-7.60.0/configure.ac curl-7.60.0.aros/configure.ac
--- curl-7.60.0/configure.ac	2018-05-14 12:01:43.000000000 +0100
+++ curl-7.60.0.aros/configure.ac	2018-06-09 06:45:42.615317739 +0100
@@ -630,6 +630,19 @@
        AC_MSG_RESULT(yes)
 )
 
+if test "$HAVE_GETHOSTBYNAME" != "1"
+then
+  dnl AROS gets to gethostbyname via bsdsocket.library
+  AC_MSG_CHECKING([for gethostbyname in bsdsocket.library])
+  AC_TRY_LINK([#include <proto/bsdsocket.h>],
+              [void *SocketBase; gethostbyname("");],
+              [ dnl worked!
+              AC_MSG_RESULT([yes])
+              HAVE_GETHOSTBYNAME="1"],
+              AC_MSG_RESULT(no)
+             )
+fi
+
 
 dnl **********************************************************************
 dnl Check for built-in manual
@@ -1715,8 +1728,23 @@
         else
             AC_MSG_RESULT(yes)
         fi
+    fi
+
+    if test "$ac_cv_lib_ssl_SSL_connect" != yes; then
+        dnl on AROS, libssl needs SocketBase
+        AC_MSG_CHECKING(for ssl with SocketBase available)
+        OLIBS=$LIBS
+        LIBS="-lssl $LIBS"
+        AC_TRY_LINK([#include <openssl/ssl.h>
+void *SocketBase;],
+                    [SSL_connect(0);],
+                    [ac_cv_lib_ssl_SSL_connect="yes"
+                     AC_MSG_RESULT([yes])],
+                    [LIBS=$OLIBS
+                     AC_MSG_RESULT([no])])
+    fi
 
-    else
+    if test "$ac_cv_lib_ssl_SSL_connect" = yes; then
 
       dnl Have the libraries--check for OpenSSL headers
       AC_CHECK_HEADERS(openssl/x509.h openssl/rsa.h openssl/crypto.h \
diff -ruN curl-7.60.0/lib/aros.c curl-7.60.0.aros/lib/aros.c
--- curl-7.60.0/lib/aros.c	1970-01-01 01:00:00.000000000 +0100
+++ curl-7.60.0.aros/lib/aros.c	2018-06-09 06:45:42.615317739 +0100
@@ -0,0 +1,69 @@
+/***************************************************************************
+ *                                  _   _ ____  _
+ *  Project                     ___| | | |  _ \| |
+ *                             / __| | | | |_) | |
+ *                            | (__| |_| |  _ <| |___
+ *                             \___|\___/|_| \_\_____|
+ *
+ * Copyright (C) 1998 - 2007, Daniel Stenberg, <daniel@haxx.se>, et al.
+ *
+ * This software is licensed as described in the file COPYING, which
+ * you should have received as part of this distribution. The terms
+ * are also available at http://curl.haxx.se/docs/copyright.html.
+ *
+ * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+ * copies of the Software, and permit persons to whom the Software is
+ * furnished to do so, under the terms of the COPYING file.
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+ * KIND, either express or implied.
+ *
+ * $Id$
+ ***************************************************************************/
+
+#ifdef __AROS__ /* AROS */
+
+#include <errno.h>
+#include <aros/symbolsets.h>
+#include <exec/libraries.h>
+#include <proto/exec.h>
+#include <proto/dos.h>
+#include <proto/bsdsocket.h>
+#include <bsdsocket/socketbasetags.h>
+
+struct Library * __attribute__((weak)) SocketBase = NULL;
+
+void aros_cleanup()
+{
+  if(SocketBase) {
+    CloseLibrary(SocketBase);
+    SocketBase = NULL;
+  }
+}
+
+BOOL aros_init()
+{
+  if(!SocketBase)
+    SocketBase = OpenLibrary("bsdsocket.library", 4);
+  
+  if(!SocketBase) {
+    Printf("No TCP/IP Stack running!", "");
+    return FALSE;
+  }
+  
+  if(SocketBaseTags(
+    SBTM_SETVAL(SBTC_ERRNOPTR(sizeof(errno))), (IPTR) &errno,
+    SBTM_SETVAL(SBTC_LOGTAGPTR),       (IPTR) "cURL",
+  TAG_DONE)) {
+    
+    Printf("SocketBaseTags ERROR", "");
+    return FALSE;
+  }
+  
+  return TRUE;
+}
+
+ADD2INIT(aros_init, -50)
+ADD2EXIT(aros_cleanup, -50)
+
+#endif /* __AROS__ */
diff -ruN curl-7.60.0/lib/connect.c curl-7.60.0.aros/lib/connect.c
--- curl-7.60.0/lib/connect.c	2018-05-07 09:20:04.000000000 +0100
+++ curl-7.60.0.aros/lib/connect.c	2018-06-09 06:45:42.615317739 +0100
@@ -58,6 +58,10 @@
 #include <inet.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include "urldata.h"
 #include "sendf.h"
 #include "if2ip.h"
diff -ruN curl-7.60.0/lib/curl_gethostname.c curl-7.60.0.aros/lib/curl_gethostname.c
--- curl-7.60.0/lib/curl_gethostname.c	2017-12-02 23:33:20.000000000 +0000
+++ curl-7.60.0.aros/lib/curl_gethostname.c	2018-06-09 06:45:42.615317739 +0100
@@ -48,6 +48,10 @@
  * For libcurl static library release builds no overriding takes place.
  */
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 int Curl_gethostname(char *name, GETHOSTNAME_TYPE_ARG2 namelen)
 {
 #ifndef HAVE_GETHOSTNAME
diff -ruN curl-7.60.0/lib/curl_setup.h curl-7.60.0.aros/lib/curl_setup.h
--- curl-7.60.0/lib/curl_setup.h	2018-05-07 10:18:02.000000000 +0100
+++ curl-7.60.0.aros/lib/curl_setup.h	2018-06-09 06:45:42.615317739 +0100
@@ -496,6 +496,13 @@
 
 #  endif /* MSDOS */
 
+#  ifdef __AROS__
+#   undef HAVE_ALARM
+#   undef HAVE_SIGNAL
+#   undef HAVE_SIGACTION
+#  endif
+
+
 #  ifdef __minix
      /* Minix 3 versions up to at least 3.1.3 are missing these prototypes */
      extern char *strtok_r(char *s, const char *delim, char **last);
diff -ruN curl-7.60.0/lib/file.c curl-7.60.0.aros/lib/file.c
--- curl-7.60.0/lib/file.c	2018-05-07 09:20:04.000000000 +0100
+++ curl-7.60.0.aros/lib/file.c	2018-06-09 06:45:42.615317739 +0100
@@ -137,7 +137,7 @@
   char *real_path;
   struct FILEPROTO *file = data->req.protop;
   int fd;
-#ifdef DOS_FILESYSTEM
+#if defined(DOS_FILESYSTEM) || defined(__AROS__)
   size_t i;
   char *actual_path;
 #endif
@@ -183,6 +183,14 @@
 
   fd = open_readonly(actual_path, O_RDONLY|O_BINARY);
   file->path = actual_path;
+#elif defined __AROS__
+  /* On AROS paths can start with volume name, skip slash in this case */
+  actual_path = real_path;
+  if(actual_path[0] == '/' && strchr(actual_path, ':'))
+    actual_path++;
+
+  fd = open(actual_path, O_RDONLY);
+  file->path = actual_path;
 #else
   if(memchr(real_path, 0, real_path_len)) {
     /* binary zeroes indicate foul play */
diff -ruN curl-7.60.0/lib/ftp.c curl-7.60.0.aros/lib/ftp.c
--- curl-7.60.0/lib/ftp.c	2018-05-14 12:30:44.000000000 +0100
+++ curl-7.60.0.aros/lib/ftp.c	2018-06-09 06:45:42.615317739 +0100
@@ -46,6 +46,10 @@
 #define in_addr_t unsigned long
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include <curl/curl.h>
 #include "urldata.h"
 #include "sendf.h"
diff -ruN curl-7.60.0/lib/getinfo.c curl-7.60.0.aros/lib/getinfo.c
--- curl-7.60.0/lib/getinfo.c	2018-05-07 09:20:04.000000000 +0100
+++ curl-7.60.0.aros/lib/getinfo.c	2018-06-09 06:45:42.615317739 +0100
@@ -31,6 +31,10 @@
 #include "connect.h" /* Curl_getconnectinfo() */
 #include "progress.h"
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 /* The last #include files should be: */
 #include "curl_memory.h"
 #include "memdebug.h"
diff -ruN curl-7.60.0/lib/hostip4.c curl-7.60.0.aros/lib/hostip4.c
--- curl-7.60.0/lib/hostip4.c	2018-05-07 09:20:04.000000000 +0100
+++ curl-7.60.0.aros/lib/hostip4.c	2018-06-09 06:45:42.615317739 +0100
@@ -45,6 +45,10 @@
 #include <process.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include "urldata.h"
 #include "sendf.h"
 #include "hostip.h"
diff -ruN curl-7.60.0/lib/http.c curl-7.60.0.aros/lib/http.c
--- curl-7.60.0/lib/http.c	2018-05-14 12:30:44.000000000 +0100
+++ curl-7.60.0.aros/lib/http.c	2018-06-09 06:45:42.615317739 +0100
@@ -45,6 +45,10 @@
 #include <sys/param.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include "urldata.h"
 #include <curl/curl.h>
 #include "transfer.h"
diff -ruN curl-7.60.0/lib/if2ip.c curl-7.60.0.aros/lib/if2ip.c
--- curl-7.60.0/lib/if2ip.c	2018-05-07 09:20:04.000000000 +0100
+++ curl-7.60.0.aros/lib/if2ip.c	2018-06-09 06:45:42.615317739 +0100
@@ -50,6 +50,10 @@
 #  include <inet.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include "inet_ntop.h"
 #include "strcase.h"
 #include "if2ip.h"
diff -ruN curl-7.60.0/lib/inet_ntop.c curl-7.60.0.aros/lib/inet_ntop.c
--- curl-7.60.0/lib/inet_ntop.c	2018-05-14 10:01:32.000000000 +0100
+++ curl-7.60.0.aros/lib/inet_ntop.c	2018-06-09 06:45:42.615317739 +0100
@@ -32,6 +32,10 @@
 #include <arpa/inet.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include "inet_ntop.h"
 #include "curl_printf.h"
 
diff -ruN curl-7.60.0/lib/Makefile.in curl-7.60.0.aros/lib/Makefile.in
--- curl-7.60.0/lib/Makefile.in	2018-05-14 13:20:31.000000000 +0100
+++ curl-7.60.0.aros/lib/Makefile.in	2018-06-09 06:45:42.615317739 +0100
@@ -227,7 +227,7 @@
 	libcurl_la-system_win32.lo libcurl_la-mime.lo \
 	libcurl_la-sha256.lo libcurl_la-setopt.lo \
 	libcurl_la-curl_path.lo libcurl_la-curl_ctype.lo \
-	libcurl_la-curl_range.lo
+	libcurl_la-curl_range.lo libcurl_la-aros.lo
 am__dirstamp = $(am__leading_dot)dirstamp
 am__objects_2 = vauth/libcurl_la-vauth.lo \
 	vauth/libcurl_la-cleartext.lo vauth/libcurl_la-cram.lo \
@@ -714,7 +714,7 @@
   http_ntlm.c curl_ntlm_wb.c curl_ntlm_core.c curl_sasl.c rand.c        \
   curl_multibyte.c hostcheck.c conncache.c pipeline.c dotdot.c          \
   x509asn1.c http2.c smb.c curl_endian.c curl_des.c system_win32.c      \
-  mime.c sha256.c setopt.c curl_path.c curl_ctype.c curl_range.c
+  mime.c sha256.c setopt.c curl_path.c curl_ctype.c curl_range.c aros.c
 
 LIB_HFILES = arpa_telnet.h netrc.h file.h timeval.h hostip.h progress.h \
   formdata.h cookie.h http.h sendf.h ftp.h url.h dict.h if2ip.h         \
@@ -971,6 +971,7 @@
 	-rm -f *.tab.c
 
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurl_la-amigaos.Plo@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurl_la-aros.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurl_la-asyn-ares.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurl_la-asyn-thread.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurl_la-base64.Plo@am__quote@
@@ -1079,6 +1080,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurl_la-wildcard.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurl_la-x509asn1.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurlu_la-amigaos.Plo@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurlu_la-aros.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurlu_la-asyn-ares.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurlu_la-asyn-thread.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libcurlu_la-base64.Plo@am__quote@
@@ -2101,6 +2103,13 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libcurl_la_CPPFLAGS) $(CPPFLAGS) $(libcurl_la_CFLAGS) $(CFLAGS) -c -o vauth/libcurl_la-spnego_sspi.lo `test -f 'vauth/spnego_sspi.c' || echo '$(srcdir)/'`vauth/spnego_sspi.c
 
+libcurl_la-aros.lo: aros.c
+@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libcurl_la_CPPFLAGS) $(CPPFLAGS) $(libcurl_la_CFLAGS) $(CFLAGS) -MT libcurl_la-aros.lo -MD -MP -MF $(DEPDIR)/libcurl_la-aros.Tpo -c -o libcurl_la-aros.lo `test -f 'aros.c' || echo '$(srcdir)/'`aros.c
+@am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/libcurl_la-aros.Tpo $(DEPDIR)/libcurl_la-aros.Plo
+@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='aros.c' object='libcurl_la-aros.lo' libtool=yes @AMDEPBACKSLASH@
+@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libcurl_la_CPPFLAGS) $(CPPFLAGS) $(libcurl_la_CFLAGS) $(CFLAGS) -c -o libcurl_la-aros.lo `test -f 'aros.c' || echo '$(srcdir)/'`aros.c
+
 vtls/libcurl_la-openssl.lo: vtls/openssl.c
 @am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libcurl_la_CPPFLAGS) $(CPPFLAGS) $(libcurl_la_CFLAGS) $(CFLAGS) -MT vtls/libcurl_la-openssl.lo -MD -MP -MF vtls/$(DEPDIR)/libcurl_la-openssl.Tpo -c -o vtls/libcurl_la-openssl.lo `test -f 'vtls/openssl.c' || echo '$(srcdir)/'`vtls/openssl.c
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) vtls/$(DEPDIR)/libcurl_la-openssl.Tpo vtls/$(DEPDIR)/libcurl_la-openssl.Plo
diff -ruN curl-7.60.0/lib/Makefile.inc curl-7.60.0.aros/lib/Makefile.inc
--- curl-7.60.0/lib/Makefile.inc	2018-05-07 10:18:02.000000000 +0100
+++ curl-7.60.0.aros/lib/Makefile.inc	2018-06-09 06:45:42.615317739 +0100
@@ -54,7 +54,7 @@
   http_ntlm.c curl_ntlm_wb.c curl_ntlm_core.c curl_sasl.c rand.c        \
   curl_multibyte.c hostcheck.c conncache.c pipeline.c dotdot.c          \
   x509asn1.c http2.c smb.c curl_endian.c curl_des.c system_win32.c      \
-  mime.c sha256.c setopt.c curl_path.c curl_ctype.c curl_range.c
+  mime.c sha256.c setopt.c curl_path.c curl_ctype.c curl_range.c aros.c
 
 LIB_HFILES = arpa_telnet.h netrc.h file.h timeval.h hostip.h progress.h \
   formdata.h cookie.h http.h sendf.h ftp.h url.h dict.h if2ip.h         \
diff -ruN curl-7.60.0/lib/memdebug.c curl-7.60.0.aros/lib/memdebug.c
--- curl-7.60.0/lib/memdebug.c	2018-05-07 09:20:04.000000000 +0100
+++ curl-7.60.0.aros/lib/memdebug.c	2018-06-09 06:45:42.615317739 +0100
@@ -26,6 +26,10 @@
 
 #include <curl/curl.h>
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include "urldata.h"
 
 #define MEMDEBUG_NODEFINES /* don't redefine the standard functions */
diff -ruN curl-7.60.0/lib/multi.c curl-7.60.0.aros/lib/multi.c
--- curl-7.60.0/lib/multi.c	2018-05-07 10:18:03.000000000 +0100
+++ curl-7.60.0.aros/lib/multi.c	2018-06-09 06:45:42.615317739 +0100
@@ -22,6 +22,10 @@
 
 #include "curl_setup.h"
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include <curl/curl.h>
 
 #include "urldata.h"
diff -ruN curl-7.60.0/lib/nonblock.c curl-7.60.0.aros/lib/nonblock.c
--- curl-7.60.0/lib/nonblock.c	2018-05-07 09:20:04.000000000 +0100
+++ curl-7.60.0.aros/lib/nonblock.c	2018-06-09 06:45:42.615317739 +0100
@@ -37,6 +37,10 @@
 #include <inet.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include "nonblock.h"
 
 /*
diff -ruN curl-7.60.0/lib/select.c curl-7.60.0.aros/lib/select.c
--- curl-7.60.0/lib/select.c	2018-05-07 09:20:04.000000000 +0100
+++ curl-7.60.0.aros/lib/select.c	2018-06-09 06:45:42.615317739 +0100
@@ -35,6 +35,10 @@
 #include <socket.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #ifdef MSDOS
 #include <dos.h>  /* delay() */
 #endif
@@ -581,3 +585,4 @@
    return rc;
 }
 #endif /* TPF */
+
diff -ruN curl-7.60.0/lib/sendf.c curl-7.60.0.aros/lib/sendf.c
--- curl-7.60.0/lib/sendf.c	2018-05-07 09:20:04.000000000 +0100
+++ curl-7.60.0.aros/lib/sendf.c	2018-06-09 06:45:42.615317739 +0100
@@ -30,6 +30,10 @@
 #include <linux/tcp.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include <curl/curl.h>
 
 #include "urldata.h"
diff -ruN curl-7.60.0/lib/telnet.c curl-7.60.0.aros/lib/telnet.c
--- curl-7.60.0/lib/telnet.c	2018-05-14 10:01:32.000000000 +0100
+++ curl-7.60.0.aros/lib/telnet.c	2018-06-09 06:45:42.615317739 +0100
@@ -44,6 +44,10 @@
 #include <sys/param.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include "urldata.h"
 #include <curl/curl.h>
 #include "transfer.h"
diff -ruN curl-7.60.0/lib/tftp.c curl-7.60.0.aros/lib/tftp.c
--- curl-7.60.0/lib/tftp.c	2018-05-07 10:18:03.000000000 +0100
+++ curl-7.60.0.aros/lib/tftp.c	2018-06-09 06:45:42.615317739 +0100
@@ -44,6 +44,10 @@
 #include <sys/param.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include "urldata.h"
 #include <curl/curl.h>
 #include "transfer.h"
diff -ruN curl-7.60.0/lib/transfer.c curl-7.60.0.aros/lib/transfer.c
--- curl-7.60.0/lib/transfer.c	2018-05-14 12:01:43.000000000 +0100
+++ curl-7.60.0.aros/lib/transfer.c	2018-06-09 06:45:42.615317739 +0100
@@ -54,6 +54,10 @@
 #error "We can't compile without socket() support!"
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include "urldata.h"
 #include <curl/curl.h>
 #include "netrc.h"
diff -ruN curl-7.60.0/lib/url.c curl-7.60.0.aros/lib/url.c
--- curl-7.60.0/lib/url.c	2018-05-14 12:30:44.000000000 +0100
+++ curl-7.60.0.aros/lib/url.c	2018-06-09 06:45:42.615317739 +0100
@@ -51,6 +51,10 @@
 #include <sys/un.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #ifndef HAVE_SOCKET
 #error "We can't compile without socket() support!"
 #endif
diff -ruN curl-7.60.0/lib/vtls/vtls.c curl-7.60.0.aros/lib/vtls/vtls.c
--- curl-7.60.0/lib/vtls/vtls.c	2018-05-14 12:01:43.000000000 +0100
+++ curl-7.60.0.aros/lib/vtls/vtls.c	2018-06-09 06:45:42.615317739 +0100
@@ -50,6 +50,11 @@
 
 #include "urldata.h"
 
+#if defined(__AROS__)
+#undef connect
+#undef shutdown
+#endif
+
 #include "vtls.h" /* generic SSL protos etc */
 #include "slist.h"
 #include "sendf.h"
diff -ruN curl-7.60.0/src/tool_main.c curl-7.60.0.aros/src/tool_main.c
--- curl-7.60.0/src/tool_main.c	2018-05-07 09:20:04.000000000 +0100
+++ curl-7.60.0.aros/src/tool_main.c	2018-06-09 06:45:42.615317739 +0100
@@ -32,6 +32,10 @@
 #include <plarenas.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #define ENABLE_CURLX_PRINTF
 /* use our own printf() functions */
 #include "curlx.h"
diff -ruN curl-7.60.0/src/tool_sleep.c curl-7.60.0.aros/src/tool_sleep.c
--- curl-7.60.0/src/tool_sleep.c	2018-05-07 09:20:04.000000000 +0100
+++ curl-7.60.0.aros/src/tool_sleep.c	2018-06-09 06:45:42.615317739 +0100
@@ -35,6 +35,10 @@
 #  include <dos.h>
 #endif
 
+#ifdef __AROS__
+#include <proto/bsdsocket.h>
+#endif
+
 #include "tool_sleep.h"
 
 #include "memdebug.h" /* keep this as LAST include */
