diff -ruN fontconfig-2.6.0/configure fontconfig-2.6.0-aros/configure
--- fontconfig-2.6.0/configure	2008-06-01 04:31:02.000000000 +0200
+++ fontconfig-2.6.0-aros/configure	2009-06-28 20:12:08.000000000 +0200
@@ -22249,76 +22249,6 @@
 
 
 #
-# Checks for FreeType
-#
-
-
-# Check whether --with-freetype-config was given.
-if test "${with_freetype_config+set}" = set; then
-  withval=$with_freetype_config; freetype_config=$withval
-else
-  freetype_config=yes
-fi
-
-
-if test "$freetype_config" = "yes"; then
-	# Extract the first word of "freetype-config", so it can be a program name with args.
-set dummy freetype-config; ac_word=$2
-{ echo "$as_me:$LINENO: checking for $ac_word" >&5
-echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6; }
-if test "${ac_cv_path_ft_config+set}" = set; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  case $ft_config in
-  [\\/]* | ?:[\\/]*)
-  ac_cv_path_ft_config="$ft_config" # Let the user override the test with a path.
-  ;;
-  *)
-  as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-  for ac_exec_ext in '' $ac_executable_extensions; do
-  if { test -f "$as_dir/$ac_word$ac_exec_ext" && $as_test_x "$as_dir/$ac_word$ac_exec_ext"; }; then
-    ac_cv_path_ft_config="$as_dir/$ac_word$ac_exec_ext"
-    echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-done
-IFS=$as_save_IFS
-
-  test -z "$ac_cv_path_ft_config" && ac_cv_path_ft_config="no"
-  ;;
-esac
-fi
-ft_config=$ac_cv_path_ft_config
-if test -n "$ft_config"; then
-  { echo "$as_me:$LINENO: result: $ft_config" >&5
-echo "${ECHO_T}$ft_config" >&6; }
-else
-  { echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6; }
-fi
-
-
-	if test "$ft_config" = "no"; then
-		{ { echo "$as_me:$LINENO: error: You must have freetype installed; see http://www.freetype.org/" >&5
-echo "$as_me: error: You must have freetype installed; see http://www.freetype.org/" >&2;}
-   { (exit 1); exit 1; }; }
-	fi
-else
-	ft_config="$freetype_config"
-fi
-
-FREETYPE_CFLAGS="`$ft_config --cflags`"
-FREETYPE_LIBS="`$ft_config --libs`"
-
-
-
-
-#
 # Check to see whether we have:
 #  FT_Get_Next_Char
 #  FT_Get_BDF_Property
@@ -22335,7 +22265,7 @@
 
 
 
-for ac_func in FT_Get_Next_Char FT_Get_BDF_Property FT_Get_PS_Font_Info FT_Has_PS_Glyph_Names FT_Get_X11_Font_Format
+for ac_func in FT_Get_Next_Char FT_Get_BDF_Property
 do
 as_ac_var=`echo "ac_cv_func_$ac_func" | $as_tr_sh`
 { echo "$as_me:$LINENO: checking for $ac_func" >&5
diff -ruN fontconfig-2.6.0/fc-cache/fc-cache.c fontconfig-2.6.0-aros/fc-cache/fc-cache.c
--- fontconfig-2.6.0/fc-cache/fc-cache.c	2008-05-04 09:34:13.000000000 +0200
+++ fontconfig-2.6.0-aros/fc-cache/fc-cache.c	2009-06-28 20:12:08.000000000 +0200
@@ -289,8 +289,8 @@
 	    continue;
 	/* skip cache files for different architectures and */
 	/* files which are not cache files at all */
-	if (strlen(ent->d_name) != 32 + strlen ("-" FC_ARCHITECTURE FC_CACHE_SUFFIX) ||
-	    strcmp(ent->d_name + 32, "-" FC_ARCHITECTURE FC_CACHE_SUFFIX))
+	if (strlen(ent->d_name) != 8 + strlen ("-" FC_ARCHITECTURE FC_CACHE_SUFFIX) ||
+	    strcmp(ent->d_name + 8, "-" FC_ARCHITECTURE FC_CACHE_SUFFIX))
 	    continue;
 	
 	file_name = FcStrPlus (dir_base, (FcChar8 *) ent->d_name);
diff -ruN fontconfig-2.6.0/fc-cache/Makefile.am fontconfig-2.6.0-aros/fc-cache/Makefile.am
--- fontconfig-2.6.0/fc-cache/Makefile.am	2008-05-04 02:20:34.000000000 +0200
+++ fontconfig-2.6.0-aros/fc-cache/Makefile.am	2009-06-28 20:12:08.000000000 +0200
@@ -30,10 +30,10 @@
 if OS_WIN32
 else
 install-data-local:
-	-$(mkinstalldirs) "$(DESTDIR)$(fc_cachedir)"
+	-$(mkinstalldirs) "$(DESTDIR)$(prefix)$(fc_cachedir)"
 
 uninstall-local:
-	-$(RM) -rf "$(DESTDIR)$(fc_cachedir)"
+	-$(RM) -rf "$(DESTDIR)$(prefix)$(fc_cachedir)"
 endif
 
 INCLUDES=-I${top_srcdir} -I${top_srcdir}/src $(WARN_CFLAGS)
diff -ruN fontconfig-2.6.0/fc-cache/Makefile.in fontconfig-2.6.0-aros/fc-cache/Makefile.in
--- fontconfig-2.6.0/fc-cache/Makefile.in	2008-06-01 04:30:58.000000000 +0200
+++ fontconfig-2.6.0-aros/fc-cache/Makefile.in	2009-06-28 20:12:08.000000000 +0200
@@ -583,10 +583,10 @@
 
 
 @OS_WIN32_FALSE@install-data-local:
-@OS_WIN32_FALSE@	-$(mkinstalldirs) "$(DESTDIR)$(fc_cachedir)"
+@OS_WIN32_FALSE@	-$(mkinstalldirs) "$(DESTDIR)$(prefix)$(fc_cachedir)"
 
 @OS_WIN32_FALSE@uninstall-local:
-@OS_WIN32_FALSE@	-$(RM) -rf "$(DESTDIR)$(fc_cachedir)"
+@OS_WIN32_FALSE@	-$(RM) -rf "$(DESTDIR)$(prefix)$(fc_cachedir)"
 
 @USEDOCBOOK_TRUE@${man_MANS}: ${SGML}	
 @USEDOCBOOK_TRUE@	$(RM) $@
diff -ruN fontconfig-2.6.0/Makefile.am fontconfig-2.6.0-aros/Makefile.am
--- fontconfig-2.6.0/Makefile.am	2008-05-25 01:55:45.000000000 +0200
+++ fontconfig-2.6.0-aros/Makefile.am	2009-06-28 20:12:08.000000000 +0200
@@ -82,7 +82,7 @@
 .PHONY: ChangeLog
 
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(configdir) $(DESTDIR)$(fc_cachedir)
+	$(mkinstalldirs) $(DESTDIR)$(configdir) $(DESTDIR)$(prefix)$(fc_cachedir)
 	if [ -f $(DESTDIR)$(configdir)/fonts.conf ]; then \
 	  echo "backing up existing $(DESTDIR)$(configdir)/fonts.conf"; \
 	  mv $(DESTDIR)$(configdir)/fonts.conf $(DESTDIR)$(configdir)/fonts.conf.bak; \
diff -ruN fontconfig-2.6.0/Makefile.in fontconfig-2.6.0-aros/Makefile.in
--- fontconfig-2.6.0/Makefile.in	2008-06-01 04:31:00.000000000 +0200
+++ fontconfig-2.6.0-aros/Makefile.in	2009-06-28 20:12:08.000000000 +0200
@@ -789,7 +789,7 @@
 .PHONY: ChangeLog
 
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(configdir) $(DESTDIR)$(fc_cachedir)
+	$(mkinstalldirs) $(DESTDIR)$(configdir) $(DESTDIR)$(prefix)$(fc_cachedir)
 	if [ -f $(DESTDIR)$(configdir)/fonts.conf ]; then \
 	  echo "backing up existing $(DESTDIR)$(configdir)/fonts.conf"; \
 	  mv $(DESTDIR)$(configdir)/fonts.conf $(DESTDIR)$(configdir)/fonts.conf.bak; \
diff -ruN fontconfig-2.6.0/src/fcatomic.c fontconfig-2.6.0-aros/src/fcatomic.c
--- fontconfig-2.6.0/src/fcatomic.c	2008-05-04 10:18:31.000000000 +0200
+++ fontconfig-2.6.0-aros/src/fcatomic.c	2009-06-28 20:12:08.000000000 +0200
@@ -104,7 +104,7 @@
     int		ret;
     struct stat	lck_stat;
 
-#ifdef HAVE_LINK
+#if defined(HAVE_LINK) && !defined(__AROS__)
     strcpy ((char *) atomic->tmp, (char *) atomic->file);
     strcat ((char *) atomic->tmp, TMP_NAME);
     fd = mkstemp ((char *) atomic->tmp);
@@ -147,7 +147,7 @@
 	    time_t  now = time (0);
 	    if ((long int) (now - lck_stat.st_mtime) > 10 * 60)
 	    {
-#ifdef HAVE_LINK
+#if defined(HAVE_LINK) && !defined(__AROS__)
 		if (unlink ((char *) atomic->lck) == 0)
 		    return FcAtomicLock (atomic);
 #else
@@ -194,7 +194,7 @@
 void
 FcAtomicUnlock (FcAtomic *atomic)
 {
-#ifdef HAVE_LINK
+#if defined(HAVE_LINK) && !defined(__AROS__)
     unlink ((char *) atomic->lck);
 #else
     rmdir ((char *) atomic->lck);
diff -ruN fontconfig-2.6.0/src/fccache.c fontconfig-2.6.0-aros/src/fccache.c
--- fontconfig-2.6.0/src/fccache.c	2007-11-14 01:36:41.000000000 +0100
+++ fontconfig-2.6.0-aros/src/fccache.c	2009-06-28 20:12:08.000000000 +0200
@@ -52,7 +52,7 @@
 static void MD5Final(unsigned char digest[16], struct MD5Context *ctx);
 static void MD5Transform(FcChar32 buf[4], FcChar32 in[16]);
 
-#define CACHEBASE_LEN (1 + 32 + 1 + sizeof (FC_ARCHITECTURE) + sizeof (FC_CACHE_SUFFIX))
+#define CACHEBASE_LEN (1 + 8 + 1 + sizeof (FC_ARCHITECTURE) + sizeof (FC_CACHE_SUFFIX))
 
 #ifdef _WIN32
 
@@ -158,7 +158,7 @@
 
     cache_base[0] = '/';
     hex_hash = cache_base + 1;
-    for (cnt = 0; cnt < 16; ++cnt)
+    for (cnt = 0; cnt < 4; ++cnt)
     {
 	hex_hash[2*cnt  ] = bin2hex[hash[cnt] >> 4];
 	hex_hash[2*cnt+1] = bin2hex[hash[cnt] & 0xf];
diff -ruN fontconfig-2.6.0/src/fccfg.c fontconfig-2.6.0-aros/src/fccfg.c
--- fontconfig-2.6.0/src/fccfg.c	2008-05-25 01:14:15.000000000 +0200
+++ fontconfig-2.6.0-aros/src/fccfg.c	2009-06-28 20:12:08.000000000 +0200
@@ -1750,6 +1750,11 @@
 	goto absolute_path;
 #endif
 
+#ifdef __AROS__
+    if (strchr(url, ':'))
+        goto absolute_path;
+#endif
+
     switch (*url) {
     case '~':
 	dir = FcConfigHome ();
@@ -1762,6 +1767,9 @@
     case '\\':
     absolute_path:
 #endif
+#ifdef __AROS__
+    absolute_path:
+#endif
     case '/':
 	file = FcConfigFileExists (0, url);
 	break;
diff -ruN fontconfig-2.6.0/src/fcint.h fontconfig-2.6.0-aros/src/fcint.h
--- fontconfig-2.6.0/src/fcint.h	2008-05-04 04:31:41.000000000 +0200
+++ fontconfig-2.6.0-aros/src/fcint.h	2009-06-28 20:12:08.000000000 +0200
@@ -54,7 +54,7 @@
 #define FC_CONFIG_PATH "fonts.conf"
 #endif
 
-#ifdef _WIN32
+#if defined _WIN32 || defined __AROS__
 #define FC_SEARCH_PATH_SEPARATOR ';'
 #else
 #define FC_SEARCH_PATH_SEPARATOR ':'
diff -ruN fontconfig-2.6.0/src/fcstr.c fontconfig-2.6.0-aros/src/fcstr.c
--- fontconfig-2.6.0/src/fcstr.c	2008-05-04 09:49:41.000000000 +0200
+++ fontconfig-2.6.0-aros/src/fcstr.c	2009-06-28 20:14:06.000000000 +0200
@@ -839,6 +839,10 @@
     FcChar8	    *slash;
 
     slash = (FcChar8 *) strrchr ((const char *) path, '/');
+#ifdef __AROS__
+    if (!slash)
+        slash = strrchr ((const char *) path, ':');
+#endif
 #ifdef _WIN32
     {
         FcChar8     *backslash;
@@ -859,6 +863,10 @@
     FcChar8 *dir;
 
     slash = FcStrLastSlash (file);
+#ifdef __AROS__
+    if (*slash == ':')
+        slash++;
+#endif
     if (!slash)
 	return FcStrCopy ((FcChar8 *) ".");
     dir = malloc ((slash - file) + 1);
@@ -976,7 +984,7 @@
     FcConvertDosPath (full);
     return FcStrCanonAbsoluteFilename (full);
 #else
-    if (s[0] == '/')
+    if (s[0] == '/' || strchr(s, ':'))
 	return FcStrCanonAbsoluteFilename (s);
     else
     {
diff -ruN fontconfig-2.6.0/src/Makefile.in fontconfig-2.6.0-aros/src/Makefile.in
--- fontconfig-2.6.0/src/Makefile.in	2008-06-01 04:31:00.000000000 +0200
+++ fontconfig-2.6.0-aros/src/Makefile.in	2009-06-28 20:12:08.000000000 +0200
@@ -118,7 +118,7 @@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
 CFLAGS = @CFLAGS@
-CONFDIR = @CONFDIR@
+CONFDIR = $(shell echo @CONFDIR@|sed 's%/%%; s%/%:%')
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CROSS_COMPILING_FALSE = @CROSS_COMPILING_FALSE@
@@ -149,8 +149,8 @@
 EXPAT_LIBS = @EXPAT_LIBS@
 F77 = @F77@
 FC_ADD_FONTS = @FC_ADD_FONTS@
-FC_CACHEDIR = @FC_CACHEDIR@
-FC_DEFAULT_FONTS = @FC_DEFAULT_FONTS@
+FC_CACHEDIR = $(shell echo @FC_CACHEDIR@|sed 's%/%%; s%/%:%')
+FC_DEFAULT_FONTS = $(shell echo @FC_DEFAULT_FONTS@|sed 's%/%%; s%/%:%')
 FC_FONTDATE = @FC_FONTDATE@
 FC_FONTPATH = @FC_FONTPATH@
 FFLAGS = @FFLAGS@
