diff -Naur abcm2ps-8.12.4-orig/abcm2ps.h abcm2ps-8.12.4/abcm2ps.h
--- abcm2ps-8.12.4-orig/abcm2ps.h	2016-07-25 10:48:10.929949686 +0200
+++ abcm2ps-8.12.4/abcm2ps.h	2016-07-25 10:49:50.960034641 +0200
@@ -3,7 +3,9 @@
 #include <stdio.h>
 #include <time.h>
 
+#ifdef HAVE_CONFIG_H
 #include "config.h"
+#endif
 
 #define MAXVOICE 32	/* max number of voices */
 
diff -Naur abcm2ps-8.12.4-orig/abcparse.c abcm2ps-8.12.4/abcparse.c
--- abcm2ps-8.12.4-orig/abcparse.c	2016-07-25 10:48:10.929949686 +0200
+++ abcm2ps-8.12.4/abcparse.c	2016-07-25 10:50:09.616162722 +0200
@@ -10,7 +10,10 @@
  * (at your option) any later version.
  */
 
+#ifdef HAVE_CONFIG_H
 #include "config.h"
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
