diff -Naur abcm2ps-8.13.1-orig/abcm2ps.h abcm2ps-8.13.1/abcm2ps.h
--- abcm2ps-8.13.1-orig/abcm2ps.h	2016-12-21 17:18:21.846383009 +0100
+++ abcm2ps-8.13.1/abcm2ps.h	2016-12-21 17:20:37.309487727 +0100
@@ -3,7 +3,9 @@
 #include <stdio.h>
 #include <time.h>
 
+#ifdef HAVE_CONFIG_H
 #include "config.h"
+#endif
 
 #define MAXVOICE 32	/* max number of voices */
 
diff -Naur abcm2ps-8.13.1-orig/abcparse.c abcm2ps-8.13.1/abcparse.c
--- abcm2ps-8.13.1-orig/abcparse.c	2016-12-21 17:18:21.846383009 +0100
+++ abcm2ps-8.13.1/abcparse.c	2016-12-21 17:20:24.503666684 +0100
@@ -10,7 +10,10 @@
  * (at your option) any later version.
  */
 
+#ifdef HAVE_CONFIG_H
 #include "config.h"
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
