diff -Naur abcm2ps-8.13.3-orig/abcm2ps.h abcm2ps-8.13.3/abcm2ps.h
--- abcm2ps-8.13.3-orig/abcm2ps.h	2017-01-02 12:12:47.300587862 +0100
+++ abcm2ps-8.13.3/abcm2ps.h	2017-01-02 12:18:14.386257737 +0100
@@ -3,7 +3,9 @@
 #include <stdio.h>
 #include <time.h>
 
+#ifdef HAVE_CONFIG_H
 #include "config.h"
+#endif
 
 #define MAXVOICE 32	/* max number of voices */
 
diff -Naur abcm2ps-8.13.3-orig/abcparse.c abcm2ps-8.13.3/abcparse.c
--- abcm2ps-8.13.3-orig/abcparse.c	2017-01-02 12:12:47.300587862 +0100
+++ abcm2ps-8.13.3/abcparse.c	2017-01-02 12:18:25.954175057 +0100
@@ -10,7 +10,10 @@
  * (at your option) any later version.
  */
 
+#ifdef HAVE_CONFIG_H
 #include "config.h"
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -Naur abcm2ps-8.13.3-orig/syms.c abcm2ps-8.13.3/syms.c
--- abcm2ps-8.13.3-orig/syms.c	2017-01-02 12:12:47.304587834 +0100
+++ abcm2ps-8.13.3/syms.c	2017-01-02 12:20:32.385270454 +0100
@@ -862,7 +862,7 @@
 				q++;
 			else
 				q = p;
-			fprintf(fout, "/music/%.*s def\n", r - q, q);
+			fprintf(fout, "/music/%.*s def\n", (int)(r - q), q);
 		}
 	}
 
