diff -Naur rocksndiamonds-4.0.0.0-orig/graphics/gfx_classic/.directory rocksndiamonds-4.0.0.0/graphics/gfx_classic/.directory
--- rocksndiamonds-4.0.0.0-orig/graphics/gfx_classic/.directory	1970-01-01 01:00:00.000000000 +0100
+++ rocksndiamonds-4.0.0.0/graphics/gfx_classic/.directory	2016-12-28 15:04:27.557438426 +0100
@@ -0,0 +1,7 @@
+[Dolphin]
+HeaderColumnWidths=277,146,124
+PreviewsShown=true
+Timestamp=2016,12,28,15,4,27
+Version=3
+ViewMode=1
+VisibleRoles=Details_text,Details_size,Details_date,Details_type,CustomizedDetails
diff -Naur rocksndiamonds-4.0.0.0-orig/src/libgame/platform.h rocksndiamonds-4.0.0.0/src/libgame/platform.h
--- rocksndiamonds-4.0.0.0-orig/src/libgame/platform.h	2016-12-28 14:37:07.948659566 +0100
+++ rocksndiamonds-4.0.0.0/src/libgame/platform.h	2016-12-28 14:50:34.837284781 +0100
@@ -41,6 +41,12 @@
 #define PLATFORM_STRING "AmigaOS"
 #endif
 
+#if defined(__AROS__)
+#define PLATFORM_AROS
+#undef  PLATFORM_STRING
+#define PLATFORM_STRING "AROS"
+#endif
+
 #if defined(__BEOS__)
 #define PLATFORM_BEOS
 #undef  PLATFORM_STRING
diff -Naur rocksndiamonds-4.0.0.0-orig/src/libgame/setup.c rocksndiamonds-4.0.0.0/src/libgame/setup.c
--- rocksndiamonds-4.0.0.0-orig/src/libgame/setup.c	2016-12-28 14:37:07.948659566 +0100
+++ rocksndiamonds-4.0.0.0/src/libgame/setup.c	2016-12-28 14:47:12.935370390 +0100
@@ -1317,6 +1317,8 @@
     if (!SUCCEEDED(SHGetFolderPath(NULL, CSIDL_PERSONAL, NULL, 0, dir)))
       strcpy(dir, ".");
   }
+#elif defined(PLATFORM_AROS)
+  dir = "/PROGDIR";
 #elif defined(PLATFORM_UNIX)
   if (dir == NULL)
   {
diff -Naur rocksndiamonds-4.0.0.0-orig/src/main.c rocksndiamonds-4.0.0.0/src/main.c
--- rocksndiamonds-4.0.0.0-orig/src/main.c	2016-12-28 14:37:07.946659567 +0100
+++ rocksndiamonds-4.0.0.0/src/main.c	2016-12-28 14:40:16.823572430 +0100
@@ -5678,11 +5678,13 @@
 	  sdl_version.minor,
 	  sdl_version.patch);
 
+#if !defined(__AROS__)
     SDL_IMAGE_VERSION(&sdl_version);
     Print("- SDL_image %d.%d.%d\n",
 	  sdl_version.major,
 	  sdl_version.minor,
 	  sdl_version.patch);
+#endif
 
     SDL_MIXER_VERSION(&sdl_version);
     Print("- SDL_mixer %d.%d.%d\n",
diff -Naur rocksndiamonds-4.0.0.0-orig/src/screens.c rocksndiamonds-4.0.0.0/src/screens.c
--- rocksndiamonds-4.0.0.0-orig/src/screens.c	2016-12-28 14:37:07.949659566 +0100
+++ rocksndiamonds-4.0.0.0/src/screens.c	2016-12-28 14:42:35.927506254 +0100
@@ -3318,8 +3318,10 @@
 	    sdl_version_linked->minor,
 	    sdl_version_linked->patch);
 
+#if !defined(__AROS__)
   SDL_IMAGE_VERSION(&sdl_version_compiled);
   sdl_version_linked = IMG_Linked_Version();
+#endif
 
   ystart2 += ystep;
   DrawTextF(xstart1, ystart2, font_text, "SDL_image");
